<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>An R interface to the Ontology Lookup Service • rols</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script><script src="../pkgdown.js"></script><!-- mathjax --><script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-vignette">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">rols</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../articles/index.html">Articles</a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><script src="rols_files/htmlwidgets-0.8/htmlwidgets.js"></script><script src="rols_files/jquery-1.12.4/jquery.min.js"></script><script src="rols_files/datatables-binding-0.2/datatables.js"></script><link href="rols_files/dt-core-1.10.12/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="rols_files/dt-core-1.10.12/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="rols_files/dt-core-1.10.12/js/jquery.dataTables.min.js"></script><div class="row">
  <div class="col-md-9">
    <div class="page-header toc-ignore">
      <h1>An R interface to the Ontology Lookup Service</h1>
                        <h4 class="author">Laurent Gatto</h4>
            <address class="author_afil">
      Computational Proteomics Unit, Cambridge, UK<br>
</address>
</div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>How to query the Ontology Lookup Service directly from R and how to create and parse controlled vocabulary.</p>
    </div>
    
<div class="contents">
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<html><body><a href="#introduction" class="anchor"> </a></body></html>Introduction</h1>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<html><body><a href="#installation" class="anchor"> </a></body></html>Installation</h2>
<p><em><a href="http://bioconductor.org/packages/rols">rols</a></em> is a Bioconductor package and should hence be installed using the dedicated functionality</p>
<pre class="sourceCode r"><code class="sourceCode r">## try http:// if https:// URLs are not supported
<span class="kw">source</span>(<span class="st">"https://bioconductor.org/biocLite.R"</span>)
<span class="kw">biocLite</span>(<span class="st">"rols"</span>)</code></pre>
</div>
<div id="getting-help" class="section level2">
<h2 class="hasAnchor">
<html><body><a href="#getting-help" class="anchor"> </a></body></html>Getting help</h2>
<p>To get help, either post your question on the <a href="https://support.bioconductor.org/">Bioconductor support site</a> or open an issue on the <em><a href="http://bioconductor.org/packages/rols">rols</a></em> <a href="https://github.com/lgatto/rols/issues">github page</a>.</p>
</div>
<div id="the-resource" class="section level2">
<h2 class="hasAnchor">
<html><body><a href="#the-resource" class="anchor"> </a></body></html>The resource</h2>
<p>The <a href="http://www.ebi.ac.uk/ontology-lookup/">Ontology Lookup Service</a> (OLS) [<a href="http://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-7-97">1</a>, <a href="http://nar.oxfordjournals.org/content/36/suppl_2/W372">2</a>] is originally spin-off of the <a href="http://www.ebi.ac.uk/pride/archive/">PRoteomics IDEntifications</a> database (PRIDE) service, located at the EBI, and is now developed and maintained by the <a href="http://www.ebi.ac.uk/about/spot-team">Samples, Phenotypes and Ontologies</a> team at EMBL-EBI.</p>
</div>
<div id="the-package" class="section level2">
<h2 class="hasAnchor">
<html><body><a href="#the-package" class="anchor"> </a></body></html>The package</h2>
<p>The OLS provides a REST interface to hundreds of ontologies from a single location with a unified output format. The <em><a href="http://bioconductor.org/packages/rols">rols</a></em> package make this possible from within R. Do do so, it relies on the <em><a href="http://cran.fhcrc.org/web/packages/httr/index.html">httr</a></em> package to query the REST interface, and access and retrieve data.</p>
<p>There are 158 ontologies available in the OLS, listed in the table below. Their name is to be use to defined which ontology to query.</p>
<div id="htmlwidget-0058b5adbecfcef5d734" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-0058b5adbecfcef5d734">{"x":{"filter":"none","data":[["agro","apo","ddpheno","dideo","dinto","dpo","dron","eco","edam","ehdaa2","emap","enm","envo","eol","epo","ero","fao","fbbi","fbbt","fbcv","fbdv","fix","flopo","flu","fma","fypo","gaz","hom","hsapdv","mmo","mro","nbo","ncro","ogi","omp","ontoneo","oostt","opl","ordo","ovae","pato","pco","pdro","pdumdv","po","pride","rexo","rnao","stato","symp","teddy","trans","tto","vario","vo","vt","wbbt","wbls","xao","ancestro","bao","bfo","bspo","caro","ceph","aeo","chebi","chmo","aero","clo","atol","cteno","bcgo","ddanat","bco","genepio","ido","bto","cco","idomal","cdao","kisao","lbo","cheminf","ma","cio","mamo","cl","cmo","mfmo","mfoem","cmpo","mi","cvdo","doid","miapa","efo","ms","oba","emapa","obib","eo","exo","ogg","foodon","probonto","pw","geo","rex","gexo","go","rxno","sio","hao","hp","swo","iao","tads","mf","tgma","mirnao","xco","miro","xl","mmusdv","mod","mop","mp","mpath","ncbitaxon","oae","obcs","obi","ogms","ogsf","olatdv","omiabis","omit","omrse","poro","pr","reto","ro","rs","sbo","sep","sibo","so","spd","taxrank","to","uberon","uo","vto","wbphenotype","zeco","zfa","zfs"],["AGRO","APO","DDPHENO","DIDEO","DINTO","FBcv","DRON","ECO","EDAM","EHDAA2","EMAP","ENM","ENVO","EOL","EPO","ERO","FAO","FBbi","FBbt","FBcv","FBdv","FIX","FLOPO","FLU","FMA","FYPO","GAZ","HOM","HSAPDV","MMO","MRO","NBO","NCRO","OGI","OMP","ONTONEO","OOSTT","OPL","ORDO","OVAE","PATO","PCO","PDRO","PDUMDV","PO","PRIDE","ReXO","RNAO","STATO","SYMP","TEDDY","TRANS","TTO","VARIO","VO","VT","WBBT","WBLS","XAO","ancestro","BAO","BFO","BSPO","CARO","CEPH","AEO","CHEBI","CHMO","AERO","CLO","ATOL","CTENO","BCGO","DDANAT","BCO","GENEPIO","IDO","BTO","CCO","IDOMAL","CDAO","KISAO","LBO","CHEMINF","MA","CIO","MAMO","CL","CMO","MFMO","MFOEM","CMPO","MI","CVDO","DOID","MIAPA","EFO","MS","OBA","EMAPA","OBIB","EO","EXO","OGG","FOODON","PROBONTO","PW","GEO","REX","GeXO","GO","RXNO","SIO","HAO","HP","SWO","IAO","TADS","MF","TGMA","MIRNAO","XCO","MIRO","XL","MMUSDV","MOD","MOP","MP","MPATH","NCBITaxon","OAE","OBCS","OBI","OGMS","OGSF","OLATDV","OMIABIS","OMIT","OMRSE","PORO","PR","ReTO","RO","RS","SBO","SEP","SIBO","SO","SPD","TAXRANK","TO","UBERON","UO","VTO","WBPHENOTYPE","ZECO","ZFA","ZFS"],["agro","apo","ddpheno","dideo","dinto","dpo","dron","eco","edam","ehdaa2","emap","enm","envo","eol","epo","ero","fao","fbbi","fbbt","fbcv","fbdv","fix","flopo","flu","fma","fypo","gaz","hom","hsapdv","mmo","mro","nbo","ncro","ogi","omp","ontoneo","oostt","opl","ordo","ovae","pato","pco","pdro","pdumdv","po","pride","rexo","rnao","stato","symp","teddy","trans","tto","vario","vo","vt","wbbt","wbls","xao","ancestro","bao","bfo","bspo","caro","ceph","aeo","chebi","chmo","aero","clo","atol","cteno","bcgo","ddanat","bco","genepio","ido","bto","cco","idomal","cdao","kisao","lbo","cheminf","ma","cio","mamo","cl","cmo","mfmo","mfoem","cmpo","mi","cvdo","doid","miapa","efo","ms","oba","emapa","obib","eo","exo","ogg","foodon","probonto","pw","geo","rex","gexo","go","rxno","sio","hao","hp","swo","iao","tads","mf","tgma","mirnao","xco","miro","xl","mmusdv","mod","mop","mp","mpath","ncbitaxon","oae","obcs","obi","ogms","ogsf","olatdv","omiabis","omit","omrse","poro","pr","reto","ro","rs","sbo","sep","sibo","so","spd","taxrank","to","uberon","uo","vto","wbphenotype","zeco","zfa","zfs"],["Agronomy Ontology","Ascomycete phenotype ontology","Dictyostelium discoideum phenotype","Drug Interaction and Evidence Ontology","The Drug-Drug Interactions Ontology","Drosophila Phenotype Ontology","The Drug Ontology","Evidence ontology","Bioinformatics operations, data types, formats, identifiers and topics","Human developmental anatomy, abstract","Mouse gross anatomy and development, timed","eNanoMapper Ontology","Environment Ontology","Environment Ontology for Livestock","Epidemiology Ontology","eagle-i resource ontology","Fungal gross anatomy","Biological imaging methods","Drosophila gross anatomy","FlyBase Controlled Vocabulary","Drosophila development","Physico-chemical methods and properties","Flora Phenotype Ontology","Influenza ontology","Foundational Model of Anatomy (subset)","Fission Yeast Phenotype Ontology","Gazetteer","Homology Ontology","Human Developmental Stages","Measurement method ontology","MHC Restriction Ontology","Neuro Behavior Ontology","Non-Coding RNA Ontology","Ontology for genetic interval","Ontology of Microbial Phenotypes","Obstetric and Neonatal Ontology","Ontology of Organizational Structures of Trauma centers and Trauma Systems","Ontology for Parasite LifeCycle","Orphanet Rare Disease Ontolog","OVAE: Ontology of Vaccine Adverse Events","Phenotypic quality","Population and Community Ontology","The Prescription of Drugs Ontology","Platynereis Developmental Stages","Plant Ontology","PRIDE Controlled Vocabulary","Regulation of Gene Expression Ontology","The RNA Ontology (RNAO): An Ontology for Integrating RNA Sequence and Structure Data","STATO: the statistical methods ontology","Symptom Ontology","Terminology for Description of Dynamics","Pathogen Transmission Ontology","Teleost taxonomy","Variation Ontology","Vaccine Ontology","Vertebrate trait","C. elegans gross anatomy","C. elegans development","Xenopus anatomy and development","Ancestry Ontology","BioAssay Ontology","Basic Formal Ontology","Biological Spatial Ontology","Common Anatomy Reference Ontology","Cephalopod Ontology","Anatomical Entity Ontology","Chemical Entities of Biological Interest","Chemical Methods Ontology","Adverse Event Reporting Ontology","CLO: Cell Line Ontology","Animal Trait Ontology for Livestock","Ctenophore Ontology","Beta Cell Genomics Ontology","Dictyostelium discoideum anatomy","Biological Collections Ontology","Genomic Epidemiology Ontology","Infectious disease","BRENDA tissue / enzyme source","Cell Cycle Ontology","Malaria Ontology","Comparative Data Analysis Ontology","Kinetic Simulation Algorithm Ontology","Livestock Breed Ontology","chemical information ontology (cheminf) - information entities about chemical entities","Mouse adult gross anatomy","Confidence Information Ontology","Mathematical Modelling Ontology","Cell Ontology","Clinical measurement ontology","Mammalian Feeding Muscle Ontology","Emotion Ontology","Cellular Microscopy Phenotype Ontology","Molecular Interactions","Cardiovascular Disease Ontology","Human Disease Ontology","Minimum Information for A Phylogenetic Analysis (MIAPA) Ontology","Experimental Factor Ontology","Mass spectrometry ontology","Ontology of Biological Attributes","Mouse gross anatomy and development, timed","Ontology for BIoBanking (OBIB)","Plant Environment Ontology","Exposure ontology","OGG: Ontology of Genes and Genomes","FOODON","Probability Distribution Ontology","Pathway ontology","Geographical Entity Ontology","Physico-chemical process","Gene Expression Ontology","Uberon Ontology Module for GO","Name Reaction Ontology","Semanticscience Integrated Ontology","Hymenoptera Anatomy Ontology","human phenotype ontology","Software ontology","Information Artifact Ontology","Tick gross anatomy","Mental Functioning Ontology","Mosquito gross anatomy","microRNA Ontology","Experimental condition ontology","Mosquito insecticide resistance","Cross-linker reagents ontology","Mouse Developmental Stages","Protein modification","Molecular Process Ontology","The Mammalian Phenotype Ontology","Mouse pathology","NCBI organismal classification","OAE: Ontology of Adverse Events","OBCS: Ontology of Biological and Clinical Statistics","Ontology for Biomedical Investigation","Ontology for General Medical Science","Ontology of Genetic Susceptibility Factors (OGSF)","Medaka Developmental Stages","Ontologized MIABIS","Ontology for MIRNA Target","Ontology of Medically Related Social Entities","Porifera (sponge) ontology","PRotein Ontology (PRO)","Regulation of Transcription Ontology","OBO Relations Ontology","Rat Strain Ontology","Systems Biology","Sample processing and separation techniques","Social Insect Behavior Ontology","Sequence types and features","Spider Ontology","Taxonomic rank vocabulary","Plant Trait Ontology","Uber-anatomy ontology","Units of measurement","Vertebrate Taxonomy Ontology","C. elegans phenotype","Zebrafish Experimental Conditions Ontology","Zebrafish anatomy and development","Zebrafish developmental stages"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> \u003c/th>\n      <th>Prefix\u003c/th>\n      <th>Namespace\u003c/th>\n      <th>Title\u003c/th>\n    \u003c/tr>\n  \u003c/thead>\n\u003c/table>","options":{"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}]}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="a-brief-rols-overview" class="section level1">
<h1 class="hasAnchor">
<html><body><a href="#a-brief-rols-overview" class="anchor"> </a></body></html>A Brief rols overview</h1>
<p>The <em><a href="http://bioconductor.org/packages/rols">rols</a></em> package is build around a few classes that enable to query the OLS and retrieve, store and manipulate data. Each of these classes are described in more details in their respective manual pages. We start by loading the package.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"rols"</span>)</code></pre>
<div id="ontologies" class="section level2">
<h2 class="hasAnchor">
<html><body><a href="#ontologies" class="anchor"> </a></body></html>Ontologies</h2>
<p>The <code>Ontology</code> and <code>Ontologies</code> classes can store information about single of multiple ontologies. The latter can be easily subset using <code>[</code> and <code>[[</code>, as one would for lists.</p>
<pre class="sourceCode r"><code class="sourceCode r">ol &lt;-<span class="st"> </span><span class="kw"><a href="../reference/Ontology-class.html">Ontologies</a></span>()
ol</code></pre>
<pre><code>## Object of class 'Ontologies' with 158 entries
##    AGRO, APO ... ZFA, ZFS</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(<span class="kw"><a href="../reference/Ontology-class.html">olsNamespace</a></span>(ol))</code></pre>
<pre><code>##      agro       apo   ddpheno     dideo     dinto       dpo 
##    "agro"     "apo" "ddpheno"   "dideo"   "dinto"     "dpo"</code></pre>
<pre class="sourceCode r"><code class="sourceCode r">ol[[<span class="st">"go"</span>]]</code></pre>
<pre><code>## Ontology: Uberon Ontology Module for GO (go)  
##   An ontology for describing the function of genes and gene products
##    Loaded: 2016-12-23 Updated: 2017-01-02 Version: 2016-12-22 
##    47900 terms  68 properties  0 individuals</code></pre>
<p>It is also possible to initialise a single ontology</p>
<pre class="sourceCode r"><code class="sourceCode r">go &lt;-<span class="st"> </span><span class="kw"><a href="../reference/Ontology-class.html">Ontology</a></span>(<span class="st">"go"</span>)
go</code></pre>
<pre><code>## Ontology: Uberon Ontology Module for GO (go)  
##   An ontology for describing the function of genes and gene products
##    Loaded: 2016-12-23 Updated: 2017-01-02 Version: 2016-12-22 
##    47900 terms  68 properties  0 individuals</code></pre>
</div>
<div id="terms" class="section level2">
<h2 class="hasAnchor">
<html><body><a href="#terms" class="anchor"> </a></body></html>Terms</h2>
<p>Single ontology terms are stored in <code>Term</code> objects. When more terms need to be manipulated, they are stored as <code>Terms</code> objects. It is easy to obtain all terms of an ontology of interest, and the resulting <code>Terms</code> object can be subset using <code>[</code> and <code>[[</code>, as one would for lists.</p>
<pre class="sourceCode r"><code class="sourceCode r">gotrms &lt;-<span class="st"> </span><span class="kw"><a href="../reference/Term-class.html">terms</a></span>(go) ## or terms("go")
gotrms</code></pre>
<pre><code>## Object of class 'Terms' with 47906 entries
##  From the GO ontology
##   GO:0005230, GO:0015276 ... GO:0032942, GO:0019987</code></pre>
<pre class="sourceCode r"><code class="sourceCode r">gotrms[<span class="dv">1</span>:<span class="dv">10</span>]</code></pre>
<pre><code>## Object of class 'Terms' with 10 entries
##  From the GO ontology
##   GO:0005230, GO:0015276 ... GO:0005261, GO:0032852</code></pre>
<pre class="sourceCode r"><code class="sourceCode r">gotrms[[<span class="st">"GO:0090575"</span>]]</code></pre>
<pre><code>## A Term from the GO ontology: GO:0090575 
##  Label: RNA polymerase II transcription factor complex
##   A transcription factor complex that acts at promoters of genes
##   transcribed by RNA polymerase II.</code></pre>
<p>It is also possible to initialise a single term</p>
<pre class="sourceCode r"><code class="sourceCode r">trm &lt;-<span class="st"> </span><span class="kw"><a href="../reference/Term-class.html">term</a></span>(go, <span class="st">"GO:0090575"</span>)
<span class="kw"><a href="../reference/Term-class.html">termId</a></span>(trm)</code></pre>
<pre><code>## [1] "GO:0090575"</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/Term-class.html">termLabel</a></span>(trm)</code></pre>
<pre><code>## [1] "RNA polymerase II transcription factor complex"</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">strwrap</span>(<span class="kw"><a href="../reference/Term-class.html">termDesc</a></span>(trm))</code></pre>
<pre><code>## [1] "A transcription factor complex that acts at promoters of genes"
## [2] "transcribed by RNA polymerase II."</code></pre>
<p>It is then possible to extract the <code>ancestors</code>, <code>descendants</code>, <code>parents</code> and <code>children</code> terms. Each of these functions return a <code>Terms</code> object</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/Term-class.html">parents</a></span>(trm)</code></pre>
<pre><code>## Object of class 'Terms' with 1 entries
##  From the GO ontology
## GO:0044798</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/Term-class.html">children</a></span>(trm)</code></pre>
<pre><code>## Object of class 'Terms' with 20 entries
##  From the GO ontology
##   GO:0005675, GO:0005674 ... GO:0097221, GO:0036488</code></pre>
<p>Similarly, the <code>partOf</code> and <code>derivesFrom</code> functions return, for an input term, the terms it is a part of and derived from.</p>
</div>
<div id="properties" class="section level2">
<h2 class="hasAnchor">
<html><body><a href="#properties" class="anchor"> </a></body></html>Properties</h2>
<p>Properties (relationships) of single or multiple terms or complete ontologies can be queries with the <code>properties</code> method, as briefly illustrated below.</p>
<pre class="sourceCode r"><code class="sourceCode r">trm &lt;-<span class="st"> </span><span class="kw"><a href="../reference/Term-class.html">term</a></span>(<span class="st">"uberon"</span>, <span class="st">"UBERON:0002107"</span>)
trm</code></pre>
<pre><code>## A Term from the UBERON ontology: UBERON:0002107 
##  Label: liver
##   An exocrine gland which secretes bile and functions in metabolism
##   of protein and carbohydrate and fat, synthesizes substances
##   involved in the clotting of the blood, synthesizes vitamin A,
##   detoxifies poisonous substances, stores glycogen, and breaks down
##   worn-out erythrocytes[GO].</code></pre>
<pre class="sourceCode r"><code class="sourceCode r">p &lt;-<span class="st"> </span><span class="kw"><a href="../reference/Properties-class.html">properties</a></span>(trm)
p</code></pre>
<pre><code>## Object of class 'Properties' with 1011 entries
##  From the UBERON ontology
##   digestive gland, abdomen element ... liver lobule, bile</code></pre>
<pre class="sourceCode r"><code class="sourceCode r">p[[<span class="dv">1</span>]]</code></pre>
<pre><code>## A Property from the UBERON ontology: UBERON:0006925 
##  Label: digestive gland</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/Term-class.html">termLabel</a></span>(p[[<span class="dv">1</span>]])</code></pre>
<pre><code>## [1] "digestive gland"</code></pre>
</div>
</div>
<div id="use-case" class="section level1">
<h1 class="hasAnchor">
<html><body><a href="#use-case" class="anchor"> </a></body></html>Use case</h1>
<p>A researcher might be interested in the trans-Golgi network. Searching the OLS is assured by the <code>OlsSearch</code> and <code>olsSearch</code> classes/functions. The first step is to defined the search query with <code>OlsSearch</code>, as shown below. This creates an search object of class <code>OlsSearch</code> that stores the query and its parameters. In records the number of requested results (default is 20) and the total number of possible results (there are 13617 results across all ontologies, in this case). At this stage, the results have not yet been downloaded, as shown by the 0 responses.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/OlsSearch-class.html">OlsSearch</a></span>(<span class="dt">q =</span> <span class="st">"trans-golgi network"</span>)</code></pre>
<pre><code>## Object of class 'OlsSearch':
##   query: trans-golgi network 
##   requested: 20 (out of 13617)
##   response(s): 0</code></pre>
<p>13617 results are probably too many to be relevant. Below we show how to perform an exact search by setting <code>exact = TRUE</code>, and limiting the search the the GO ontology by specifying <code>ontology = "GO"</code>, or doing both.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/OlsSearch-class.html">OlsSearch</a></span>(<span class="dt">q =</span> <span class="st">"trans-golgi network"</span>, <span class="dt">exact =</span> <span class="ot">TRUE</span>)</code></pre>
<pre><code>## Object of class 'OlsSearch':
##   query: trans-golgi network 
##   requested: 20 (out of 7)
##   response(s): 0</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/OlsSearch-class.html">OlsSearch</a></span>(<span class="dt">q =</span> <span class="st">"trans-golgi network"</span>, <span class="dt">ontology =</span> <span class="st">"GO"</span>)</code></pre>
<pre><code>## Object of class 'OlsSearch':
##   ontolgy: GO 
##   query: trans-golgi network 
##   requested: 20 (out of 675)
##   response(s): 0</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/OlsSearch-class.html">OlsSearch</a></span>(<span class="dt">q =</span> <span class="st">"trans-golgi network"</span>, <span class="dt">ontology =</span> <span class="st">"GO"</span>, <span class="dt">exact =</span> <span class="ot">TRUE</span>)</code></pre>
<pre><code>## Object of class 'OlsSearch':
##   ontolgy: GO 
##   query: trans-golgi network 
##   requested: 20 (out of 1)
##   response(s): 0</code></pre>
<p>One case set the <code>rows</code> argument to set the number of desired results.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/OlsSearch-class.html">OlsSearch</a></span>(<span class="dt">q =</span> <span class="st">"trans-golgi network"</span>, <span class="dt">ontology =</span> <span class="st">"GO"</span>, <span class="dt">rows =</span> <span class="dv">200</span>)</code></pre>
<pre><code>## Object of class 'OlsSearch':
##   ontolgy: GO 
##   query: trans-golgi network 
##   requested: 200 (out of 675)
##   response(s): 0</code></pre>
<p>Alternatively, one can call the <code>allRows</code> function to request all results.</p>
<pre class="sourceCode r"><code class="sourceCode r">(tgnq &lt;-<span class="st"> </span><span class="kw"><a href="../reference/OlsSearch-class.html">OlsSearch</a></span>(<span class="dt">q =</span> <span class="st">"trans-golgi network"</span>, <span class="dt">ontology =</span> <span class="st">"GO"</span>))</code></pre>
<pre><code>## Object of class 'OlsSearch':
##   ontolgy: GO 
##   query: trans-golgi network 
##   requested: 20 (out of 675)
##   response(s): 0</code></pre>
<pre class="sourceCode r"><code class="sourceCode r">(tgnq &lt;-<span class="st"> </span><span class="kw"><a href="../reference/OlsSearch-class.html">allRows</a></span>(tgnq))</code></pre>
<pre><code>## Object of class 'OlsSearch':
##   ontolgy: GO 
##   query: trans-golgi network 
##   requested: 675 (out of 675)
##   response(s): 0</code></pre>
<p>Let’s proceed with the exact search and retrieve the results. Even if we request the default 20 results, only the 7 relevant result will be retrieved. The <code>olsSearch</code> function updates the previously created object (called <code>qry</code> below) by adding the results to it.</p>
<pre class="sourceCode r"><code class="sourceCode r">qry &lt;-<span class="st"> </span><span class="kw"><a href="../reference/OlsSearch-class.html">OlsSearch</a></span>(<span class="dt">q =</span> <span class="st">"trans-golgi network"</span>, <span class="dt">exact =</span> <span class="ot">TRUE</span>)
(qry &lt;-<span class="st"> </span><span class="kw"><a href="../reference/OlsSearch-class.html">olsSearch</a></span>(qry))</code></pre>
<pre><code>## Object of class 'OlsSearch':
##   query: trans-golgi network 
##   requested: 20 (out of 7)
##   response(s): 7</code></pre>
<p>We can now transform this search result object into a fully fledged <code>Terms</code> object or a <code>data.frame</code>.</p>
<pre class="sourceCode r"><code class="sourceCode r">(qtrms &lt;-<span class="st"> </span><span class="kw">as</span>(qry, <span class="st">"Terms"</span>))</code></pre>
<pre><code>## Object of class 'Terms' with 7 entries
##  From  7 ontologies
##   OMIT:0020822, GO:0005802 ... GO:0005802, GO:0005802</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">str</span>(qdrf &lt;-<span class="st"> </span><span class="kw">as</span>(qry, <span class="st">"data.frame"</span>))</code></pre>
<pre><code>## 'data.frame':    7 obs. of  10 variables:
##  $ id                  : chr  "omit:http://purl.obolibrary.org/obo/OMIT_0020822" "go:http://purl.obolibrary.org/obo/GO_0005802" "cco:http://purl.obolibrary.org/obo/GO_0005802" "nbo:http://purl.obolibrary.org/obo/GO_0005802" ...
##  $ iri                 : chr  "http://purl.obolibrary.org/obo/OMIT_0020822" "http://purl.obolibrary.org/obo/GO_0005802" "http://purl.obolibrary.org/obo/GO_0005802" "http://purl.obolibrary.org/obo/GO_0005802" ...
##  $ short_form          : chr  "OMIT_0020822" "GO_0005802" "GO_0005802" "GO_0005802" ...
##  $ obo_id              : chr  "OMIT:0020822" "GO:0005802" "GO:0005802" "GO:0005802" ...
##  $ label               : chr  "trans-Golgi Network" "trans-Golgi network" "trans-Golgi network" "trans-Golgi network" ...
##  $ ontology_name       : chr  "omit" "go" "cco" "nbo" ...
##  $ ontology_prefix     : chr  "OMIT" "GO" "CCO" "NBO" ...
##  $ type                : chr  "class" "class" "class" "class" ...
##  $ is_defining_ontology: logi  TRUE TRUE FALSE FALSE FALSE FALSE ...
##  $ description         :List of 7
##   ..$ : NULL
##   ..$ : chr "The network of interconnected tubular and cisternal structures located within the Golgi apparatus on the side d"| __truncated__
##   ..$ : chr "The network of interconnected tubular and cisternal structures located within the Golgi apparatus on the side d"| __truncated__
##   ..$ : chr "The network of interconnected tubular and cisternal structures located within the Golgi apparatus on the side d"| __truncated__
##   ..$ : chr "The network of interconnected tubular and cisternal structures located within the Golgi apparatus on the side d"| __truncated__
##   ..$ : chr "The network of interconnected tubular and cisternal structures located within the Golgi apparatus on the side d"| __truncated__
##   ..$ : chr "The network of interconnected tubular and cisternal structures located within the Golgi apparatus on the side d"| __truncated__</code></pre>
<p>In this case, we can see that we actually retrieve the same term used across different ontologies. In such cases, it might be useful to keep only non-redundant term instances. In this case, this would have been equivalent to searching the omit, go ontology</p>
<pre class="sourceCode r"><code class="sourceCode r">qtrms &lt;-<span class="st"> </span><span class="kw">unique</span>(qtrms)
<span class="kw"><a href="../reference/Term-class.html">termOntology</a></span>(qtrms)</code></pre>
<pre><code>## OMIT:0020822   GO:0005802 
##       "omit"         "go"</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/Term-class.html">termNamespace</a></span>(qtrms)</code></pre>
<pre><code>## $`OMIT:0020822`
## NULL
## 
## $`GO:0005802`
## [1] "cellular_component"</code></pre>
<p>Below, we execute the same query using the <em><a href="http://bioconductor.org/packages/GO.db">GO.db</a></em> package.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"GO.db"</span>)
GOTERM[[<span class="st">"GO:0005802"</span>]]</code></pre>
<pre><code>## GOID: GO:0005802
## Term: trans-Golgi network
## Ontology: CC
## Definition: The network of interconnected tubular and cisternal
##     structures located within the Golgi apparatus on the side
##     distal to the endoplasmic reticulum, from which secretory
##     vesicles emerge. The trans-Golgi network is important in the
##     later stages of protein secretion where it is thought to play
##     a key role in the sorting and targeting of secreted proteins
##     to the correct destination.
## Synonym: Golgi trans face
## Synonym: Golgi trans-face
## Synonym: late Golgi
## Synonym: maturing face
## Synonym: TGN
## Synonym: trans face
## Synonym: trans Golgi network</code></pre>
</div>
<div id="on-line-vs-off-line-data" class="section level1">
<h1 class="hasAnchor">
<html><body><a href="#on-line-vs-off-line-data" class="anchor"> </a></body></html>On-line vs. off-line data</h1>
<p>It is possible to observe different results with <em><a href="http://bioconductor.org/packages/rols">rols</a></em> and <em><a href="http://bioconductor.org/packages/GO.db">GO.db</a></em>, as a result of the different ways they access the data. <em><a href="http://bioconductor.org/packages/rols">rols</a></em> or <em><a href="http://bioconductor.org/packages/biomaRt">biomaRt</a></em> perform direct online queries, while <em><a href="http://bioconductor.org/packages/GO.db">GO.db</a></em> and other annotation packages use database snapshot that are updated every release.</p>
<p>Both approaches have advantages. While online queries allow to obtain the latest up-to-date information, such approaches rely on network availability and quality. If reproducibility is a major issue, the version of the database to be queried can easily be controlled with off-line approaches. In the case of <em><a href="http://bioconductor.org/packages/rols">rols</a></em>, although the load date of a specific ontology can be queried with <code>olsVersion</code>, it is not possible to query a specific version of an ontology.</p>
</div>
<div id="changes-in-version-2-0" class="section level1">
<h1 class="hasAnchor">
<html><body><a href="#changes-in-version-2-0" class="anchor"> </a></body></html>Changes in version 2.0</h1>
<p><em><a href="http://bioconductor.org/packages/rols">rols</a></em> 2.0 has substantially changed. While the table below shows some correspondence between the old and new interface, this is not always the case. The new interface relies on the <code>Ontology</code>/<code>Ontologies</code>, <code>Term</code>/<code>Terms</code> and <code>OlsSearch</code> classes, that need to be instantiated and can then be queried, as described above.</p>
<table class="table">
<thead><tr class="header">
<th align="left">version &lt; 1.99</th>
<th align="left">version &gt;= 1.99</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left"><code>ontologyLoadDate</code></td>
<td align="left">
<code>olsLoaded</code> and <code>olsUpdated</code>
</td>
</tr>
<tr class="even">
<td align="left"><code>ontologyNames</code></td>
<td align="left"><code>Ontologies</code></td>
</tr>
<tr class="odd">
<td align="left"><code>olsVersion</code></td>
<td align="left"><code>olsVersion</code></td>
</tr>
<tr class="even">
<td align="left"><code>allIds</code></td>
<td align="left"><code>terms</code></td>
</tr>
<tr class="odd">
<td align="left"><code>isIdObsolete</code></td>
<td align="left"><code>isObsolete</code></td>
</tr>
<tr class="even">
<td align="left"><code>rootId</code></td>
<td align="left"><code>olsRoot</code></td>
</tr>
<tr class="odd">
<td align="left"><code>olsQuery</code></td>
<td align="left">
<code>OlsSearch</code> and <code>olsSearch</code>
</td>
</tr>
</tbody>
</table>
<p>Not all functionality is currently available. If there is anything that you need but not available in the new version, please contact the maintained by opening an <a href="https://github.com/lgatto/rols/issues">issue</a> on the package development site.</p>
</div>
<div id="cvparams" class="section level1">
<h1 class="hasAnchor">
<html><body><a href="#cvparams" class="anchor"> </a></body></html>CVParams</h1>
<p>The <code>CVParam</code> class is used to handle controlled vocabulary. It can be used for user-defined parameters</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/CVParam-class.html">CVParam</a></span>(<span class="dt">name =</span> <span class="st">"A user param"</span>, <span class="dt">value =</span> <span class="st">"the value"</span>)</code></pre>
<pre><code>## [, , A user param, the value]</code></pre>
<p>or official controlled vocabulary (which triggers a query to the OLS service)</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/CVParam-class.html">CVParam</a></span>(<span class="dt">label =</span> <span class="st">"MS"</span>, <span class="dt">accession =</span> <span class="st">"MS:1000073"</span>)</code></pre>
<pre><code>## [MS, MS:1000073, electrospray ionization, ]</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/CVParam-class.html">CVParam</a></span>(<span class="dt">label =</span> <span class="st">"MS"</span>, <span class="dt">name =</span><span class="st">"electrospray ionization"</span>)</code></pre>
<pre><code>## [MS, MS:1000073, electrospray ionization, ]</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/CVParam-class.html">CVParam</a></span>(<span class="dt">label =</span> <span class="st">"MS"</span>, <span class="dt">name =</span><span class="st">"ESI"</span>) ## using a synonym</code></pre>
<pre><code>## [MS, MS:1000073, ESI, ]</code></pre>
<p>See <code><a href="../reference/CVParam-class.html">?CVParam</a></code> for more details and examples.</p>
</div>
<div id="session-information" class="section level1">
<h1 class="hasAnchor">
<html><body><a href="#session-information" class="anchor"> </a></body></html>Session information</h1>
<pre><code>## R Under development (unstable) (2016-12-09 r71771)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 14.04.5 LTS
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] DT_0.2               rols_2.3.2           GO.db_3.4.0         
##  [4] AnnotationDbi_1.37.0 IRanges_2.9.14       S4Vectors_0.13.5    
##  [7] Biobase_2.35.0       BiocGenerics_0.21.1  BiocStyle_2.3.28    
## [10] pkgdown_0.1.0.9000  
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_0.12.8.3      msdata_0.15.0      compiler_3.4.0    
##  [4] prettyunits_1.0.2  progress_1.1.2     tools_3.4.0       
##  [7] digest_0.6.10      pkgload_0.0.0.9000 jsonlite_1.1      
## [10] evaluate_0.10      RSQLite_1.1-1      memoise_1.0.0     
## [13] tibble_1.2         DBI_0.5-1          curl_2.3          
## [16] yaml_2.1.14        withr_1.0.2        stringr_1.1.0     
## [19] httr_1.2.1         roxygen2_5.0.1     knitr_1.15.1      
## [22] htmlwidgets_0.8    desc_1.0.1         rprojroot_1.1     
## [25] R6_2.2.0           rmarkdown_1.3      purrr_0.2.2       
## [28] magrittr_1.5       whisker_0.3-2      backports_1.0.4   
## [31] htmltools_0.3.5    assertthat_0.1     stringi_1.1.2     
## [34] lazyeval_0.2.0     crayon_1.3.2</code></pre>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#introduction">Introduction</a><ul class="nav nav-pills nav-stacked">
<li><a href="#installation">Installation</a></li>
      <li><a href="#getting-help">Getting help</a></li>
      <li><a href="#the-resource">The resource</a></li>
      <li><a href="#the-package">The package</a></li>
      </ul>
</li>
      <li>
<a href="#a-brief-rols-overview">A Brief rols overview</a><ul class="nav nav-pills nav-stacked">
<li><a href="#ontologies">Ontologies</a></li>
      <li><a href="#terms">Terms</a></li>
      <li><a href="#properties">Properties</a></li>
      </ul>
</li>
      <li><a href="#use-case">Use case</a></li>
      <li><a href="#on-line-vs.off-line-data">On-line vs. off-line data</a></li>
      <li><a href="#changes-in-version-2.0">Changes in version 2.0</a></li>
      <li><a href="#cvparams">CVParams</a></li>
      <li><a href="#session-information">Session information</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Laurent Gatto.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  </body>
</html>
