<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>An R interface to the Ontology Lookup Service • rols</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script><script src="../pkgdown.js"></script><!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-vignette">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">rols</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../articles/index.html">Articles</a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><script src="rols_files/htmlwidgets-0.9/htmlwidgets.js"></script><script src="rols_files/jquery-1.12.4/jquery.min.js"></script><script src="rols_files/datatables-binding-0.2/datatables.js"></script><link href="rols_files/dt-core-1.10.12/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="rols_files/dt-core-1.10.12/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="rols_files/dt-core-1.10.12/js/jquery.dataTables.min.js"></script><div class="row">
  <div class="col-md-9">
    <div class="page-header toc-ignore">
      <h1>An R interface to the Ontology Lookup Service</h1>
                        <h4 class="author">Laurent Gatto</h4>
            <address class="author_afil">
      Computational Proteomics Unit, Cambridge, UK<br>
</address>
</div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>How to query the Ontology Lookup Service directly from R and how to create and parse controlled vocabulary.</p>
    </div>
    
<div class="contents">
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p><em><a href="http://bioconductor.org/packages/rols">rols</a></em> is a Bioconductor package and should hence be installed using the dedicated functionality</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## try http:// if https:// URLs are not supported
<span class="kw">source</span>(<span class="st">"https://bioconductor.org/biocLite.R"</span>)
<span class="kw">biocLite</span>(<span class="st">"rols"</span>)</code></pre></div>
</div>
<div id="getting-help" class="section level2">
<h2 class="hasAnchor">
<a href="#getting-help" class="anchor"></a>Getting help</h2>
<p>To get help, either post your question on the <a href="https://support.bioconductor.org/">Bioconductor support site</a> or open an issue on the <em><a href="http://bioconductor.org/packages/rols">rols</a></em> <a href="https://github.com/lgatto/rols/issues">github page</a>.</p>
</div>
<div id="the-resource" class="section level2">
<h2 class="hasAnchor">
<a href="#the-resource" class="anchor"></a>The resource</h2>
<p>The <a href="http://www.ebi.ac.uk/ontology-lookup/">Ontology Lookup Service</a> (OLS) [<a href="http://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-7-97">1</a>, <a href="http://nar.oxfordjournals.org/content/36/suppl_2/W372">2</a>] is originally spin-off of the <a href="http://www.ebi.ac.uk/pride/archive/">PRoteomics IDEntifications</a> database (PRIDE) service, located at the EBI, and is now developed and maintained by the <a href="http://www.ebi.ac.uk/about/spot-team">Samples, Phenotypes and Ontologies</a> team at EMBL-EBI.</p>
</div>
<div id="the-package" class="section level2">
<h2 class="hasAnchor">
<a href="#the-package" class="anchor"></a>The package</h2>
<p>The OLS provides a REST interface to hundreds of ontologies from a single location with a unified output format. The <em><a href="http://bioconductor.org/packages/rols">rols</a></em> package make this possible from within R. Do do so, it relies on the <em><a href="https://CRAN.R-project.org/package=httr">httr</a></em> package to query the REST interface, and access and retrieve data.</p>
<p>There are 196 ontologies available in the OLS, listed in the table below. Their name is to be use to defined which ontology to query.</p>
<div id="htmlwidget-bd4badf0d9cb196b8309" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-bd4badf0d9cb196b8309">{"x":{"filter":"none","data":[["aeo","ancestro","apo","bao","bco","bfo","bspo","caro","cdao","chmo","cio","clo","cmpo","co_321","co_323","co_325","co_330","co_335","co_337","co_339","co_341","co_345","co_346","co_356","dideo","dinto","eco","emap","enm","epo","ero","fao","fbcv","fypo","genepio","geno","geo","gexo","go","hao","hom","lbo","mamo","mfmo","mi","miapa","mirnao","miro","mmo","mmusdv","mod","mondo","mpath","mro","ms","ncit","oae","oarcs","obcs","ogsf","ohd","olatdv","omiabis","omit","oostt","pco","pdro","pdumdv","plana","po","ppo","pride","pw","reto","rex","ro","rs","sbo","sep","sibo","sio","spd","to","trans","uberon","vt","wbls","wbphenotype","xco","zfa","zfs","aero","agro","atol","bcgo","bto","cco","ceph","chebi","cheminf","cl","cmo","co_320","co_322","co_324","co_327","co_331","co_334","co_336","co_338","co_340","co_343","co_347","co_350","co_357","cro","cteno","cvdo","ddanat","ddpheno","doid","dpo","dron","duo","edam","efo","ehdaa2","emapa","envo","eo","eol","exo","fbbi","fbbt","fbdv","fix","flopo","flu","fma","foodon","gaz","hp","hsapdv","iao","ico","ido","idomal","kisao","ma","mf","mfoem","mop","mp","nbo","ncbitaxon","ncro","oba","obi","obib","ogg","ogi","ogms","ohmi","omp","omrse","ontoneo","opl","ordo","ovae","pato","poro","pr","probonto","rexo","rnao","rxno","scdo","sepio","so","stato","swo","symp","tads","taxrank","teddy","tgma","tto","unimod","uo","vario","vo","vto","wbbt","xao","xl","zeco"],["AEO","ancestro","APO","BAO","BCO","BFO","BSPO","CARO","CDAO","CHMO","CIO","CLO","CMPO","CO_321","CO_323","CO_325","CO_330","CO_335","CO_337","CO_339","CO_341","CO_345","CO_346","CO_356","DIDEO","DINTO","ECO","EMAP","ENM","EPO","ERO","FAO","FBcv","FYPO","GENEPIO","GENO","GEO","GeXO","GO","HAO","HOM","LBO","MAMO","MFMO","MI","MIAPA","MIRNAO","MIRO","MMO","MMUSDV","MOD","MONDO","MPATH","MRO","MS","NCIT","OAE","OARCS","OBCS","OGSF","OHD","OLATDV","OMIABIS","OMIT","OOSTT","PCO","PDRO","PDUMDV","PLANA","PO","PPO","PRIDE","PW","ReTO","REX","RO","RS","SBO","SEP","SIBO","SIO","SPD","TO","TRANS","UBERON","VT","WBLS","WBPHENOTYPE","XCO","ZFA","ZFS","AERO","AGRO","ATOL","BCGO","BTO","CCO","CEPH","CHEBI","CHEMINF","CL","CMO","CO_320","CO_322","CO_324","CO_327","CO_331","CO_334","CO_336","CO_338","CO_340","CO_343","CO_347","CO_350","CO_357","CRO","CTENO","CVDO","DDANAT","DDPHENO","DOID","FBcv","DRON","DUO","EDAM","EFO","EHDAA2","EMAPA","ENVO","EO","EOL","EXO","FBbi","FBbt","FBdv","FIX","FLOPO","FLU","FMA","FOODON","GAZ","HP","HSAPDV","IAO","ICO","IDO","IDOMAL","KISAO","MA","MF","MFOEM","MOP","MP","NBO","NCBITaxon","NCRO","OBA","OBI","OBIB","OGG","OGI","OGMS","OHMI","OMP","OMRSE","ONTONEO","OPL","ORDO","OVAE","PATO","PORO","PR","PROBONTO","ReXO","RNAO","RXNO","SCDO","SEPIO","SO","STATO","SWO","SYMP","TADS","TAXRANK","TEDDY","TGMA","TTO","UNIMOD","UO","VARIO","VO","VTO","WBBT","XAO","XL","ZECO"],["aeo","ancestro","apo","bao","bco","bfo","bspo","caro","cdao","chmo","cio","clo","cmpo","co_321","co_323","co_325","co_330","co_335","co_337","co_339","co_341","co_345","co_346","co_356","dideo","dinto","eco","emap","enm","epo","ero","fao","fbcv","fypo","genepio","geno","geo","gexo","go","hao","hom","lbo","mamo","mfmo","mi","miapa","mirnao","miro","mmo","mmusdv","mod","mondo","mpath","mro","ms","ncit","oae","oarcs","obcs","ogsf","ohd","olatdv","omiabis","omit","oostt","pco","pdro","pdumdv","plana","po","ppo","pride","pw","reto","rex","ro","rs","sbo","sep","sibo","sio","spd","to","trans","uberon","vt","wbls","wbphenotype","xco","zfa","zfs","aero","agro","atol","bcgo","bto","cco","ceph","chebi","cheminf","cl","cmo","co_320","co_322","co_324","co_327","co_331","co_334","co_336","co_338","co_340","co_343","co_347","co_350","co_357","cro","cteno","cvdo","ddanat","ddpheno","doid","dpo","dron","duo","edam","efo","ehdaa2","emapa","envo","eo","eol","exo","fbbi","fbbt","fbdv","fix","flopo","flu","fma","foodon","gaz","hp","hsapdv","iao","ico","ido","idomal","kisao","ma","mf","mfoem","mop","mp","nbo","ncbitaxon","ncro","oba","obi","obib","ogg","ogi","ogms","ohmi","omp","omrse","ontoneo","opl","ordo","ovae","pato","poro","pr","probonto","rexo","rnao","rxno","scdo","sepio","so","stato","swo","symp","tads","taxrank","teddy","tgma","tto","unimod","uo","vario","vo","vto","wbbt","xao","xl","zeco"],["Anatomical Entity Ontology","Ancestry Ontology","Ascomycete phenotype ontology","BioAssay Ontology","Biological Collections Ontology","Basic Formal Ontology","Biological Spatial Ontology","Common Anatomy Reference Ontology","Comparative Data Analysis Ontology","Chemical Methods Ontology","Confidence Information Ontology","CLO: Cell Line Ontology","Cellular Microscopy Phenotype Ontology","Wheat ontology","Barley Trait Dictionary ontology","Banana ontology","Potato ontology","Common bean ontology","Groundnut ontology","Lentil ontology","Pigeonpea ontology","Brachiaria ontology","Mungbean ontology","Vitis ontology","Drug Interaction and Evidence Ontology","The Drug-Drug Interactions Ontology","Evidence ontology","Mouse gross anatomy and development, timed","eNanoMapper Ontology","Epidemiology Ontology","eagle-i resource ontology","Fungal gross anatomy","FlyBase Controlled Vocabulary","Fission Yeast Phenotype Ontology","Genomic Epidemiology Ontology","Genotype Ontology","Geographical Entity Ontology","Gene Expression Ontology","Gene Ontology","Hymenoptera Anatomy Ontology","Homology Ontology","Livestock Breed Ontology","Mathematical Modelling Ontology","Mammalian Feeding Muscle Ontology","Molecular interactions","Minimum Information for A Phylogenetic Analysis (MIAPA) Ontology","microRNA Ontology","Mosquito insecticide resistance","Measurement method ontology","Mouse Developmental Stages","Protein modification","Monarch Disease Ontology","Mouse pathology","MHC Restriction Ontology","Mass spectrometry ontology","NCI Thesaurus OBO Edition","OAE: Ontology of Adverse Events","Ontology of Arthropod Circulatory Systems","Ontology of Biological and Clinical Statistics","Ontology of Genetic Susceptibility Factors (OGSF)","The Oral Health and Disease Ontology","Medaka Developmental Stages","Ontologized MIABIS","Ontology for MIRNA Target","Ontology of Organizational Structures of Trauma centers and Trauma Systems","Population and Community Ontology","The Prescription of Drugs Ontology","Platynereis Developmental Stages","planaria-ontology","Plant Ontology","Plant Phenology Ontology","PRIDE Controlled Vocabulary","Pathway ontology","Regulation of Transcription Ontology","Physico-chemical process","OBO Relations Ontology","Rat Strain Ontology","Systems Biology","Sample processing and separation techniques","Social Insect Behavior Ontology","Semanticscience Integrated Ontology","Spider Ontology","Plant Trait Ontology","Pathogen Transmission Ontology","Uber-anatomy ontology","Vertebrate trait","C. elegans development","C. elegans phenotype","Experimental condition ontology","Zebrafish anatomy and development","Zebrafish developmental stages","Adverse Event Reporting Ontology","Agronomy Ontology","Animal Trait Ontology for Livestock","Beta Cell Genomics Ontology","BRENDA tissue / enzyme source","Cell Cycle Ontology","Cephalopod Ontology","Chemical Entities of Biological Interest","chemical information ontology (cheminf) - information entities about chemical entities","Cell Ontology","Clinical measurement ontology","Rice ontology","Maize ontology","Sorghum ontology","Pearl millet ontology","Sweet Potato ontology","Cassava ontology","Soybean ontology","Chickpea ontology","Cowpea ontology","Yam ontology","Castor bean ontology","Oat ontology","Woody Plant Ontology ontology","Contributor Role Ontology","Ctenophore Ontology","Cardiovascular Disease Ontology","Dictyostelium discoideum anatomy","Dictyostelium discoideum phenotype","Human Disease Ontology","Drosophila Phenotype Ontology","The Drug Ontology","The Data Use Ontology","Bioinformatics operations, data types, formats, identifiers and topics","Experimental Factor Ontology","Human developmental anatomy, abstract","Mouse gross anatomy and development, timed","Environment Ontology","Plant Environment Ontology","Environment Ontology for Livestock","Exposure ontology","Biological imaging methods","Drosophila gross anatomy","Drosophila development","Physico-chemical methods and properties","Flora Phenotype Ontology","Influenza ontology","Foundational Model of Anatomy (subset)","FOODON","Gazetteer","human phenotype ontology","Human Developmental Stages","Information Artifact Ontology","Informed Consent Ontology","Infectious disease","Malaria Ontology","Kinetic Simulation Algorithm Ontology","Mouse adult gross anatomy","Mental Functioning Ontology","Emotion Ontology","Molecular Process Ontology","The Mammalian Phenotype Ontology","Neuro Behavior Ontology","NCBI organismal classification","Non-Coding RNA Ontology","Ontology of Biological Attributes","Ontology for Biomedical Investigation","Ontology for BIoBanking (OBIB)","OGG: Ontology of Genes and Genomes","Ontology for genetic interval","Ontology for General Medical Science","OHMI: Ontology of Host-Microbiome Interactions","Ontology of Microbial Phenotypes","Ontology of Medically Related Social Entities","Obstetric and Neonatal Ontology","Ontology for Parasite LifeCycle","Orphanet Rare Disease Ontolog","OVAE: Ontology of Vaccine Adverse Events","Phenotypic quality","Porifera (sponge) ontology","PRotein Ontology (PRO)","Probability Distribution Ontology","Regulation of Gene Expression Ontology","The RNA Ontology (RNAO): An Ontology for Integrating RNA Sequence and Structure Data","Name Reaction Ontology","Sickle Cell Disease Ontology","Scientific Evidence and Provenance Information Ontology","Sequence types and features","STATO: the statistical methods ontology","Bioinformatics operations, data types, formats, identifiers and topics","Symptom Ontology","Tick gross anatomy","Taxonomic rank vocabulary","Terminology for Description of Dynamics","Mosquito gross anatomy","Teleost taxonomy","Unimod protein modification database for mass spectrometry","Units of measurement","Variation Ontology","Vaccine Ontology","Vertebrate Taxonomy Ontology","C. elegans gross anatomy","Xenopus anatomy and development","Cross-linker reagents ontology","Zebrafish Experimental Conditions Ontology"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Prefix<\/th>\n      <th>Namespace<\/th>\n      <th>Title<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}]}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="a-brief-rols-overview" class="section level1">
<h1 class="hasAnchor">
<a href="#a-brief-rols-overview" class="anchor"></a>A Brief rols overview</h1>
<p>The <em><a href="http://bioconductor.org/packages/rols">rols</a></em> package is build around a few classes that enable to query the OLS and retrieve, store and manipulate data. Each of these classes are described in more details in their respective manual pages. We start by loading the package.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"rols"</span>)</code></pre></div>
<div id="ontologies" class="section level2">
<h2 class="hasAnchor">
<a href="#ontologies" class="anchor"></a>Ontologies</h2>
<p>The <code>Ontology</code> and <code>Ontologies</code> classes can store information about single of multiple ontologies. The latter can be easily subset using <code>[</code> and <code>[[</code>, as one would for lists.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ol &lt;-<span class="st"> </span><span class="kw"><a href="../reference/Ontology-class.html">Ontologies</a></span>()
ol</code></pre></div>
<pre><code>## Object of class 'Ontologies' with 196 entries
##    AEO, ancestro ... XL, ZECO</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(<span class="kw"><a href="../reference/Ontology-class.html">olsNamespace</a></span>(ol))</code></pre></div>
<pre><code>##        aeo   ancestro        apo        bao        bco        bfo 
##      "aeo" "ancestro"      "apo"      "bao"      "bco"      "bfo"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ol[[<span class="st">"go"</span>]]</code></pre></div>
<pre><code>## Ontology: Gene Ontology (go)  
##   An ontology for describing the function of genes and gene products
##    Loaded: 2017-08-31 Updated: 2017-09-01 Version: 2017-08-30 
##    49075 terms  67 properties  0 individuals</code></pre>
<p>It is also possible to initialise a single ontology</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">go &lt;-<span class="st"> </span><span class="kw"><a href="../reference/Ontology-class.html">Ontology</a></span>(<span class="st">"go"</span>)
go</code></pre></div>
<pre><code>## Ontology: Gene Ontology (go)  
##   An ontology for describing the function of genes and gene products
##    Loaded: 2017-08-31 Updated: 2017-09-01 Version: 2017-08-30 
##    49075 terms  67 properties  0 individuals</code></pre>
</div>
<div id="terms" class="section level2">
<h2 class="hasAnchor">
<a href="#terms" class="anchor"></a>Terms</h2>
<p>Single ontology terms are stored in <code>Term</code> objects. When more terms need to be manipulated, they are stored as <code>Terms</code> objects. It is easy to obtain all terms of an ontology of interest, and the resulting <code>Terms</code> object can be subset using <code>[</code> and <code>[[</code>, as one would for lists.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">gotrms &lt;-<span class="st"> </span><span class="kw"><a href="../reference/Term-class.html">terms</a></span>(go) ## or terms("go")
gotrms</code></pre></div>
<pre><code>## Object of class 'Terms' with 49076 entries
##  From the GO ontology
##   GO:0005230, GO:0015276 ... GO:0032942, GO:0019987</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">gotrms[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>]</code></pre></div>
<pre><code>## Object of class 'Terms' with 10 entries
##  From the GO ontology
##   GO:0005230, GO:0015276 ... GO:0032852, GO:0005234</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">gotrms[[<span class="st">"GO:0090575"</span>]]</code></pre></div>
<pre><code>## A Term from the GO ontology: GO:0090575 
##  Label: RNA polymerase II transcription factor complex
##   A transcription factor complex that acts at promoters of genes
##   transcribed by RNA polymerase II.</code></pre>
<p>It is also possible to initialise a single term</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">trm &lt;-<span class="st"> </span><span class="kw"><a href="../reference/Term-class.html">term</a></span>(go, <span class="st">"GO:0090575"</span>)
<span class="kw"><a href="../reference/Term-class.html">termId</a></span>(trm)</code></pre></div>
<pre><code>## [1] "GO:0090575"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/Term-class.html">termLabel</a></span>(trm)</code></pre></div>
<pre><code>## [1] "RNA polymerase II transcription factor complex"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">strwrap</span>(<span class="kw"><a href="../reference/Term-class.html">termDesc</a></span>(trm))</code></pre></div>
<pre><code>## [1] "A transcription factor complex that acts at promoters of genes"
## [2] "transcribed by RNA polymerase II."</code></pre>
<p>It is then possible to extract the <code>ancestors</code>, <code>descendants</code>, <code>parents</code> and <code>children</code> terms. Each of these functions return a <code>Terms</code> object</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/Term-class.html">parents</a></span>(trm)</code></pre></div>
<pre><code>## Object of class 'Terms' with 1 entries
##  From the GO ontology
## GO:0044798</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/Term-class.html">children</a></span>(trm)</code></pre></div>
<pre><code>## Object of class 'Terms' with 20 entries
##  From the GO ontology
##   GO:0005675, GO:0005674 ... GO:0097221, GO:0036488</code></pre>
<p>Similarly, the <code>partOf</code> and <code>derivesFrom</code> functions return, for an input term, the terms it is a part of and derived from.</p>
</div>
<div id="properties" class="section level2">
<h2 class="hasAnchor">
<a href="#properties" class="anchor"></a>Properties</h2>
<p>Properties (relationships) of single or multiple terms or complete ontologies can be queries with the <code>properties</code> method, as briefly illustrated below.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">trm &lt;-<span class="st"> </span><span class="kw"><a href="../reference/Term-class.html">term</a></span>(<span class="st">"uberon"</span>, <span class="st">"UBERON:0002107"</span>)
trm</code></pre></div>
<pre><code>## A Term from the UBERON ontology: UBERON:0002107 
##  Label: liver
##   An exocrine gland which secretes bile and functions in metabolism
##   of protein and carbohydrate and fat, synthesizes substances
##   involved in the clotting of the blood, synthesizes vitamin A,
##   detoxifies poisonous substances, stores glycogen, and breaks down
##   worn-out erythrocytes[GO].</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p &lt;-<span class="st"> </span><span class="kw"><a href="../reference/Properties-class.html">properties</a></span>(trm)
p</code></pre></div>
<pre><code>## Object of class 'Properties' with 116 entries
##  From the UBERON ontology
##   digestive gland, abdomen element ... hepatobiliary system, bile</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p[[<span class="dv">1</span>]]</code></pre></div>
<pre><code>## A Property from the UBERON ontology: UBERON:0006925 
##  Label: digestive gland</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/Term-class.html">termLabel</a></span>(p[[<span class="dv">1</span>]])</code></pre></div>
<pre><code>## [1] "digestive gland"</code></pre>
</div>
</div>
<div id="use-case" class="section level1">
<h1 class="hasAnchor">
<a href="#use-case" class="anchor"></a>Use case</h1>
<p>A researcher might be interested in the trans-Golgi network. Searching the OLS is assured by the <code>OlsSearch</code> and <code>olsSearch</code> classes/functions. The first step is to defined the search query with <code>OlsSearch</code>, as shown below. This creates an search object of class <code>OlsSearch</code> that stores the query and its parameters. In records the number of requested results (default is 20) and the total number of possible results (there are 12420 results across all ontologies, in this case). At this stage, the results have not yet been downloaded, as shown by the 0 responses.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/OlsSearch-class.html">OlsSearch</a></span>(<span class="dt">q =</span> <span class="st">"trans-golgi network"</span>)</code></pre></div>
<pre><code>## Object of class 'OlsSearch':
##   query: trans-golgi network 
##   requested: 20 (out of 12420)
##   response(s): 0</code></pre>
<p>12420 results are probably too many to be relevant. Below we show how to perform an exact search by setting <code>exact = TRUE</code>, and limiting the search the the GO ontology by specifying <code>ontology = "GO"</code>, or doing both.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/OlsSearch-class.html">OlsSearch</a></span>(<span class="dt">q =</span> <span class="st">"trans-golgi network"</span>, <span class="dt">exact =</span> <span class="ot">TRUE</span>)</code></pre></div>
<pre><code>## Object of class 'OlsSearch':
##   query: trans-golgi network 
##   requested: 20 (out of 7)
##   response(s): 0</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/OlsSearch-class.html">OlsSearch</a></span>(<span class="dt">q =</span> <span class="st">"trans-golgi network"</span>, <span class="dt">ontology =</span> <span class="st">"GO"</span>)</code></pre></div>
<pre><code>## Object of class 'OlsSearch':
##   ontolgy: GO 
##   query: trans-golgi network 
##   requested: 20 (out of 736)
##   response(s): 0</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/OlsSearch-class.html">OlsSearch</a></span>(<span class="dt">q =</span> <span class="st">"trans-golgi network"</span>, <span class="dt">ontology =</span> <span class="st">"GO"</span>, <span class="dt">exact =</span> <span class="ot">TRUE</span>)</code></pre></div>
<pre><code>## Object of class 'OlsSearch':
##   ontolgy: GO 
##   query: trans-golgi network 
##   requested: 20 (out of 1)
##   response(s): 0</code></pre>
<p>One case set the <code>rows</code> argument to set the number of desired results.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/OlsSearch-class.html">OlsSearch</a></span>(<span class="dt">q =</span> <span class="st">"trans-golgi network"</span>, <span class="dt">ontology =</span> <span class="st">"GO"</span>, <span class="dt">rows =</span> <span class="dv">200</span>)</code></pre></div>
<pre><code>## Object of class 'OlsSearch':
##   ontolgy: GO 
##   query: trans-golgi network 
##   requested: 200 (out of 736)
##   response(s): 0</code></pre>
<p>Alternatively, one can call the <code>allRows</code> function to request all results.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">(tgnq &lt;-<span class="st"> </span><span class="kw"><a href="../reference/OlsSearch-class.html">OlsSearch</a></span>(<span class="dt">q =</span> <span class="st">"trans-golgi network"</span>, <span class="dt">ontology =</span> <span class="st">"GO"</span>))</code></pre></div>
<pre><code>## Object of class 'OlsSearch':
##   ontolgy: GO 
##   query: trans-golgi network 
##   requested: 20 (out of 736)
##   response(s): 0</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">(tgnq &lt;-<span class="st"> </span><span class="kw"><a href="../reference/OlsSearch-class.html">allRows</a></span>(tgnq))</code></pre></div>
<pre><code>## Object of class 'OlsSearch':
##   ontolgy: GO 
##   query: trans-golgi network 
##   requested: 736 (out of 736)
##   response(s): 0</code></pre>
<p>Let’s proceed with the exact search and retrieve the results. Even if we request the default 20 results, only the 7 relevant result will be retrieved. The <code>olsSearch</code> function updates the previously created object (called <code>qry</code> below) by adding the results to it.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">qry &lt;-<span class="st"> </span><span class="kw"><a href="../reference/OlsSearch-class.html">OlsSearch</a></span>(<span class="dt">q =</span> <span class="st">"trans-golgi network"</span>, <span class="dt">exact =</span> <span class="ot">TRUE</span>)
(qry &lt;-<span class="st"> </span><span class="kw"><a href="../reference/OlsSearch-class.html">olsSearch</a></span>(qry))</code></pre></div>
<pre><code>## Object of class 'OlsSearch':
##   query: trans-golgi network 
##   requested: 20 (out of 7)
##   response(s): 7</code></pre>
<p>We can now transform this search result object into a fully fledged <code>Terms</code> object or a <code>data.frame</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">(qtrms &lt;-<span class="st"> </span><span class="kw">as</span>(qry, <span class="st">"Terms"</span>))</code></pre></div>
<pre><code>## Object of class 'Terms' with 7 entries
##  From  7 ontologies
##   NULL, OMIT:0020822 ... GO:0005802, GO:0005802</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">str</span>(qdrf &lt;-<span class="st"> </span><span class="kw">as</span>(qry, <span class="st">"data.frame"</span>))</code></pre></div>
<pre><code>## 'data.frame':    7 obs. of  10 variables:
##  $ id                  : chr  "ncit:class:http://purl.obolibrary.org/obo/NCIT_C33802" "omit:http://purl.obolibrary.org/obo/OMIT_0020822" "go:class:http://purl.obolibrary.org/obo/GO_0005802" "cco:http://purl.obolibrary.org/obo/GO_0005802" ...
##  $ iri                 : chr  "http://purl.obolibrary.org/obo/NCIT_C33802" "http://purl.obolibrary.org/obo/OMIT_0020822" "http://purl.obolibrary.org/obo/GO_0005802" "http://purl.obolibrary.org/obo/GO_0005802" ...
##  $ short_form          : chr  "NCIT_C33802" "OMIT_0020822" "GO_0005802" "GO_0005802" ...
##  $ label               : chr  "Trans-Golgi Network" "trans-Golgi Network" "trans-Golgi network" "trans-Golgi network" ...
##  $ description         :List of 7
##   ..$ : chr "A network of membrane components where vesicles bud off the Golgi apparatus to bring proteins, membranes and ot"| __truncated__
##   ..$ : NULL
##   ..$ : chr "The network of interconnected tubular and cisternal structures located within the Golgi apparatus on the side d"| __truncated__
##   ..$ : chr "The network of interconnected tubular and cisternal structures located within the Golgi apparatus on the side d"| __truncated__
##   ..$ : chr "The network of interconnected tubular and cisternal structures located within the Golgi apparatus on the side d"| __truncated__
##   ..$ : chr "The network of interconnected tubular and cisternal structures located within the Golgi apparatus on the side d"| __truncated__
##   ..$ : chr "The network of interconnected tubular and cisternal structures located within the Golgi apparatus on the side d"| __truncated__
##  $ ontology_name       : chr  "ncit" "omit" "go" "cco" ...
##  $ ontology_prefix     : chr  "NCIT" "OMIT" "GO" "CCO" ...
##  $ type                : chr  "class" "class" "class" "class" ...
##  $ is_defining_ontology: logi  TRUE TRUE TRUE FALSE FALSE FALSE ...
##  $ obo_id              : chr  NA "OMIT:0020822" "GO:0005802" "GO:0005802" ...</code></pre>
<p>In this case, we can see that we actually retrieve the same term used across different ontologies. In such cases, it might be useful to keep only non-redundant term instances. Here, this would have been equivalent to searching the ncit, omit, go ontology</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">qtrms &lt;-<span class="st"> </span><span class="kw">unique</span>(qtrms)
<span class="kw"><a href="../reference/Term-class.html">termOntology</a></span>(qtrms)</code></pre></div>
<pre><code>##  NCIT:C33802 OMIT:0020822   GO:0005802 
##       "ncit"       "omit"         "go"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/Term-class.html">termNamespace</a></span>(qtrms)</code></pre></div>
<pre><code>## $`NCIT:C33802`
## NULL
## 
## $`OMIT:0020822`
## NULL
## 
## $`GO:0005802`
## [1] "cellular_component"</code></pre>
<p>Below, we execute the same query using the <em><a href="http://bioconductor.org/packages/GO.db">GO.db</a></em> package.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"GO.db"</span>)
GOTERM[[<span class="st">"GO:0005802"</span>]]</code></pre></div>
<pre><code>## GOID: GO:0005802
## Term: trans-Golgi network
## Ontology: CC
## Definition: The network of interconnected tubular and cisternal
##     structures located within the Golgi apparatus on the side
##     distal to the endoplasmic reticulum, from which secretory
##     vesicles emerge. The trans-Golgi network is important in the
##     later stages of protein secretion where it is thought to play
##     a key role in the sorting and targeting of secreted proteins
##     to the correct destination.
## Synonym: Golgi trans face
## Synonym: Golgi trans-face
## Synonym: late Golgi
## Synonym: maturing face
## Synonym: TGN
## Synonym: trans face
## Synonym: trans Golgi network</code></pre>
</div>
<div id="on-line-vs-off-line-data" class="section level1">
<h1 class="hasAnchor">
<a href="#on-line-vs-off-line-data" class="anchor"></a>On-line vs. off-line data</h1>
<p>It is possible to observe different results with <em><a href="http://bioconductor.org/packages/rols">rols</a></em> and <em><a href="http://bioconductor.org/packages/GO.db">GO.db</a></em>, as a result of the different ways they access the data. <em><a href="http://bioconductor.org/packages/rols">rols</a></em> or <em><a href="http://bioconductor.org/packages/biomaRt">biomaRt</a></em> perform direct online queries, while <em><a href="http://bioconductor.org/packages/GO.db">GO.db</a></em> and other annotation packages use database snapshot that are updated every release.</p>
<p>Both approaches have advantages. While online queries allow to obtain the latest up-to-date information, such approaches rely on network availability and quality. If reproducibility is a major issue, the version of the database to be queried can easily be controlled with off-line approaches. In the case of <em><a href="http://bioconductor.org/packages/rols">rols</a></em>, although the load date of a specific ontology can be queried with <code>olsVersion</code>, it is not possible to query a specific version of an ontology.</p>
</div>
<div id="changes-in-version-2-0" class="section level1">
<h1 class="hasAnchor">
<a href="#changes-in-version-2-0" class="anchor"></a>Changes in version 2.0</h1>
<p><em><a href="http://bioconductor.org/packages/rols">rols</a></em> 2.0 has substantially changed. While the table below shows some correspondence between the old and new interface, this is not always the case. The new interface relies on the <code>Ontology</code>/<code>Ontologies</code>, <code>Term</code>/<code>Terms</code> and <code>OlsSearch</code> classes, that need to be instantiated and can then be queried, as described above.</p>
<table class="table">
<thead><tr class="header">
<th>version &lt; 1.99</th>
<th>version &gt;= 1.99</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><code>ontologyLoadDate</code></td>
<td>
<code>olsLoaded</code> and <code>olsUpdated</code>
</td>
</tr>
<tr class="even">
<td><code>ontologyNames</code></td>
<td><code>Ontologies</code></td>
</tr>
<tr class="odd">
<td><code>olsVersion</code></td>
<td><code>olsVersion</code></td>
</tr>
<tr class="even">
<td><code>allIds</code></td>
<td><code>terms</code></td>
</tr>
<tr class="odd">
<td><code>isIdObsolete</code></td>
<td><code>isObsolete</code></td>
</tr>
<tr class="even">
<td><code>rootId</code></td>
<td><code>olsRoot</code></td>
</tr>
<tr class="odd">
<td><code>olsQuery</code></td>
<td>
<code>OlsSearch</code> and <code>olsSearch</code>
</td>
</tr>
</tbody>
</table>
<p>Not all functionality is currently available. If there is anything that you need but not available in the new version, please contact the maintained by opening an <a href="https://github.com/lgatto/rols/issues">issue</a> on the package development site.</p>
</div>
<div id="cvparams" class="section level1">
<h1 class="hasAnchor">
<a href="#cvparams" class="anchor"></a>CVParams</h1>
<p>The <code>CVParam</code> class is used to handle controlled vocabulary. It can be used for user-defined parameters</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/CVParam-class.html">CVParam</a></span>(<span class="dt">name =</span> <span class="st">"A user param"</span>, <span class="dt">value =</span> <span class="st">"the value"</span>)</code></pre></div>
<pre><code>## [, , A user param, the value]</code></pre>
<p>or official controlled vocabulary (which triggers a query to the OLS service)</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/CVParam-class.html">CVParam</a></span>(<span class="dt">label =</span> <span class="st">"MS"</span>, <span class="dt">accession =</span> <span class="st">"MS:1000073"</span>)</code></pre></div>
<pre><code>## [MS, MS:1000073, electrospray ionization, ]</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/CVParam-class.html">CVParam</a></span>(<span class="dt">label =</span> <span class="st">"MS"</span>, <span class="dt">name =</span><span class="st">"electrospray ionization"</span>)</code></pre></div>
<pre><code>## [MS, MS:1000073, electrospray ionization, ]</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/CVParam-class.html">CVParam</a></span>(<span class="dt">label =</span> <span class="st">"MS"</span>, <span class="dt">name =</span><span class="st">"ESI"</span>) ## using a synonym</code></pre></div>
<pre><code>## [MS, MS:1000073, ESI, ]</code></pre>
<p>See <code><a href="../reference/CVParam-class.html">?CVParam</a></code> for more details and examples.</p>
</div>
<div id="session-information" class="section level1">
<h1 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session information</h1>
<pre><code>## R version 3.4.0 Patched (2017-06-12 r72786)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 14.04.5 LTS
## 
## Matrix products: default
## BLAS: /usr/lib/atlas-base/libf77blas.so.3.0
## LAPACK: /usr/lib/lapack/liblapack.so.3.0
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
## [1] DT_0.2               rols_2.5.2           BiocStyle_2.5.19    
## [4] GO.db_3.4.1          AnnotationDbi_1.39.3 IRanges_2.11.12     
## [7] S4Vectors_0.15.5     Biobase_2.37.2       BiocGenerics_0.23.1 
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_0.12.12      knitr_1.17        magrittr_1.5     
##  [4] progress_1.1.2    bit_1.1-12        R6_2.2.2         
##  [7] rlang_0.1.2       httr_1.3.1        stringr_1.2.0    
## [10] blob_1.1.0        tools_3.4.0       DBI_0.7          
## [13] htmltools_0.3.6   assertthat_0.2.0  yaml_2.1.14      
## [16] bit64_0.9-7       rprojroot_1.2     digest_0.6.12    
## [19] tibble_1.3.3      htmlwidgets_0.9   curl_2.8.1       
## [22] memoise_1.1.0     evaluate_0.10.1   RSQLite_2.0      
## [25] rmarkdown_1.6     stringi_1.1.5     compiler_3.4.0   
## [28] prettyunits_1.0.2 backports_1.1.0   jsonlite_1.5     
## [31] pkgconfig_2.0.1</code></pre>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#introduction">Introduction</a><ul class="nav nav-pills nav-stacked">
<li><a href="#installation">Installation</a></li>
      <li><a href="#getting-help">Getting help</a></li>
      <li><a href="#the-resource">The resource</a></li>
      <li><a href="#the-package">The package</a></li>
      </ul>
</li>
      <li>
<a href="#a-brief-rols-overview">A Brief rols overview</a><ul class="nav nav-pills nav-stacked">
<li><a href="#ontologies">Ontologies</a></li>
      <li><a href="#terms">Terms</a></li>
      <li><a href="#properties">Properties</a></li>
      </ul>
</li>
      <li><a href="#use-case">Use case</a></li>
      <li><a href="#on-line-vs.off-line-data">On-line vs. off-line data</a></li>
      <li><a href="#changes-in-version-2.0">Changes in version 2.0</a></li>
      <li><a href="#cvparams">CVParams</a></li>
      <li><a href="#session-information">Session information</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Laurent Gatto.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  </body>
</html>
