<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>An R interface to the Ontology Lookup Service • rols</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="An R interface to the Ontology Lookup Service">
<meta property="og:description" content="rols">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">rols</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/rols.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/lgatto/rols/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="rols_files/htmlwidgets-1.6.4/htmlwidgets.js"></script><link href="rols_files/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="rols_files/datatables-binding-0.31/datatables.js"></script><link href="rols_files/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="rols_files/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="rols_files/dt-core-1.13.6/js/jquery.dataTables.min.js"></script><link href="rols_files/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet">
<script src="rols_files/crosstalk-1.2.1/js/crosstalk.min.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>An R interface to the Ontology Lookup
Service</h1>
                        <h4 data-toc-skip class="author">Laurent
Gatto</h4>
            <address class="author_afil">
      <a href="https://lgatto.github.io/about/" class="external-link uri">https://lgatto.github.io/about/</a><br><small class="dont-index">Source: <a href="https://github.com/lgatto/rols/blob/HEAD/vignettes/rols.Rmd" class="external-link"><code>vignettes/rols.Rmd</code></a></small>
      <div class="hidden name"><code>rols.Rmd</code></div>

    </address>
</div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>How to query the Ontology Lookup Service directly from R and
      how to create and parse controlled vocabulary.</p>
    </div>
    
<pre><code><span><span class="co">## ⠙ Iterating 5 done (2.3/s) | 2.2s</span></span></code></pre>
<pre><code><span><span class="co">## ⠹ Iterating 10 done (3.1/s) | 3.2s</span></span></code></pre>
<pre><code><span><span class="co">## ⠹ Iterating 13 done (3.5/s) | 3.7s</span></span></code></pre>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<div class="section level3">
<h3 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
<p><em><a href="https://bioconductor.org/packages/3.19/rols" class="external-link">rols</a></em> is a
Bioconductor package and should hence be installed using the dedicated
functionality</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## try http:// if https:// URLs are not supported</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"rols"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="getting-help">Getting help<a class="anchor" aria-label="anchor" href="#getting-help"></a>
</h3>
<p>To get help, either post your question on the <a href="https://support.bioconductor.org/" class="external-link">Bioconductor support site</a>
or open an issue on the <em><a href="https://bioconductor.org/packages/3.19/rols" class="external-link">rols</a></em> <a href="https://github.com/lgatto/rols/issues" class="external-link">github page</a>.</p>
</div>
<div class="section level3">
<h3 id="the-resource">The resource<a class="anchor" aria-label="anchor" href="#the-resource"></a>
</h3>
<p>The <a href="http://www.ebi.ac.uk/ontology-lookup/" class="external-link">Ontology Lookup
Service</a> (OLS) [<a href="http://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-7-97" class="external-link">1</a>,
<a href="http://nar.oxfordjournals.org/content/36/suppl_2/W372" class="external-link">2</a>]
is originally spin-off of the <a href="http://www.ebi.ac.uk/pride/archive/" class="external-link">PRoteomics
IDEntifications</a> database (PRIDE) service, located at the EBI, and is
now developed and maintained by the <a href="http://www.ebi.ac.uk/about/spot-team" class="external-link">Samples, Phenotypes and
Ontologies</a> team at EMBL-EBI.</p>
</div>
<div class="section level3">
<h3 id="the-package">The package<a class="anchor" aria-label="anchor" href="#the-package"></a>
</h3>
<p>The OLS provides a REST interface to hundreds of ontologies from a
single location with a unified output format. The <em><a href="https://bioconductor.org/packages/3.19/rols" class="external-link">rols</a></em> package
make this possible from within R. Do do so, it relies on the <em><a href="https://CRAN.R-project.org/package=httr" class="external-link">httr</a></em> package to
query the REST interface, and access and retrieve data.</p>
<p>There are 251 ontologies available in the OLS, listed in the table
below. Their name is to be use to defined which ontology to query.</p>
<div class="datatables html-widget html-fill-item" id="htmlwidget-9c47cd2bf75e82c8e8d7" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-9c47cd2bf75e82c8e8d7">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137","138","139","140","141","142","143","144","145","146","147","148","149","150","151","152","153","154","155","156","157","158","159","160","161","162","163","164","165","166","167","168","169","170","171","172","173","174","175","176","177","178","179","180","181","182","183","184","185","186","187","188","189","190","191","192","193","194","195","196","197","198","199","200","201","202","203","204","205","206","207","208","209","210","211","212","213","214","215","216","217","218","219","220","221","222","223","224","225","226","227","228","229","230","231","232","233","234","235","236","237","238","239","240","241","242","243","244","245","246","247","248","249","250","251"],["ADO","AGRO","AISM","AMPHX","APO","APOLLO_SV","ARO","BCO","BFO","BSPO","BTO","CARO","CDAO","CDNO","CHEBI","CHEMINF","CHIRO","CHMO","CIDO","CIO","CL","CLAO","CLO","CLYH","CMO","COB","COLAO","CRO","CTENO","DDANAT","DDPHENO","DIDEO","DISDRIV","DOID","FBcv","DRON","DUO","ECAO","ECO","ECOCORE","ECTO","EMAPA","ENVO","EPIO","ExO","FAO","FBbt","FBcv","FBdv","FIDEO","FLOPO","FOBI","FOODON","FOVT","FYPO","GECKO","GENEPIO","GENO","GEO","GNO","GO","HANCESTRO","HAO","HOM","HsapDv","HSO","HTN","IAO","ICEO","ICO","IDO","INO","LABO","LEPAO","MA","MAXO","MCO","MCRO","MF","MFOEM","MFOMD","MI","MIAPA","MMO","MmusDv","MOD","MONDO","MOP","MP","MPATH","MPIO","MRO","MS","NBO","NCBITaxon","NCIT","NCRO","NGBO","NOMEN","OAE","OARCS","OBA","OBCS","OBI","OBIB","OCCO","OGG","OGMS","OHD","OHMI","OHPI","OMIT","OMO","OMP","OMRSE","ONE","ONS","ONTOAVIDA","ONTONEO","OOSTT","OPL","OPMI","ORNASEQ","OVAE","PATO","PCL","PCO","PDRO","PHIPO","PLANA","PLANP","PO","PORO","PPO","PR","PROCO","PSDO","PSO","PW","RBO","RO","RS","RXNO","SEPIO","SO","SPD","STATO","SWO","SYMP","T4FS","TAXRANK","TO","TRANS","TTO","TXPO","UBERON","UO","VBO","VO","VT","VTO","WBbt","WBls","WBPhenotype","XAO","XCO","XLMOD","XPO","ZECO","ZFA","ZFS","ZP","GSSO","HP","KISAO","SBO","SCDO","","MAMO","","SIBO","EUPATH","FBbi","MFMO","MICRO","OGSF","AEO","","","FMA","","","","","","","","","","","","","","","semapv","SLM","LIPIDMAPS","","CAO","","","","","","","","MSIO","DICOM","SDGIO","ENSGLOSS","EFO","ORDO","TEDDY","EDAM","CCO","ReXO","ReTO","GeXO","CMPO","ENM","PROBONTO","PRIDE","BAO","SIO","UNIMOD","NMR","AFO","SRAO","HCAO","IDO-COVID-19","BCIO","COVOC","OM","REPR","PROV","ORTH","CRYOEM","SNOMED","SHARELOC","CCF","CPONT"],["ado","agro","aism","amphx","apo","apollo_sv","aro","bco","bfo","bspo","bto","caro","cdao","cdno","chebi","cheminf","chiro","chmo","cido","cio","cl","clao","clo","clyh","cmo","cob","colao","cro","cteno","ddanat","ddpheno","dideo","disdriv","doid","dpo","dron","duo","ecao","eco","ecocore","ecto","emapa","envo","epio","exo","fao","fbbt","fbcv","fbdv","fideo","flopo","fobi","foodon","fovt","fypo","gecko","genepio","geno","geo","gno","go","hancestro","hao","hom","hsapdv","hso","htn","iao","iceo","ico","ido","ino","labo","lepao","ma","maxo","mco","mcro","mf","mfoem","mfomd","mi","miapa","mmo","mmusdv","mod","mondo","mop","mp","mpath","mpio","mro","ms","nbo","ncbitaxon","ncit","ncro","ngbo","nomen","oae","oarcs","oba","obcs","obi","obib","occo","ogg","ogms","ohd","ohmi","ohpi","omit","omo","omp","omrse","one","ons","ontoavida","ontoneo","oostt","opl","opmi","ornaseq","ovae","pato","pcl","pco","pdro","phipo","plana","planp","po","poro","ppo","pr","proco","psdo","pso","pw","rbo","ro","rs","rxno","sepio","so","spd","stato","swo","symp","t4fs","taxrank","to","trans","tto","txpo","uberon","uo","vbo","vo","vt","vto","wbbt","wbls","wbphenotype","xao","xco","xlmod","xpo","zeco","zfa","zfs","zp","gsso","hp","kisao","sbo","scdo","fix","mamo","rex","sibo","eupath","fbbi","mfmo","micro","ogsf","aeo","ceph","ehdaa2","fma","gaz","idomal","miro","tads","tgma","upa","aero","bcgo","bila","emap","fbsp","omiabis","tao","vsao","semapv","slm","lipidmaps","oio","cao","rdfs","owl","skos","dcterms","dc","schemaorg_http","schemaorg_https","msio","dicom","sdgio","ensemblglossary","efo","ordo","teddy","edam","cco","rexo","reto","gexo","cmpo","enm","probonto","pride","bao","sio","unimod","nmrcv","afo","srao","hcao","idocovid19","bcio","covoc","om","reproduceme","prov","orth","cryoem","snomed","shareloc","ccf","cpont"],["Alzheimer's Disease Ontology (ADO)","Agronomy Ontology","Ontology for the Anatomy of the Insect SkeletoMuscular system","Amphioxus Development and Anatomy Ontology (AMPHX)","Ascomycete Phenotype Ontology (APO)","Apollo Structured Vocabulary (Apollo-SV)","Antibiotic Resistance Ontology","Biological Collections Ontology","Basic Formal Ontology","Biological Spatial Ontology","The BRENDA Tissue Ontology (BTO)","Common Anatomy Reference Ontology","Comparative Data Analysis Ontology","Compositional Dietary Nutrition Ontology","Chemical Entities of Biological Interest","chemical information ontology (cheminf) - information entities about chemical entities","CHEBI Integrated Role Ontology","Chemical Methods Ontology","CIDO: Ontology of Coronavirus Infectious Disease","Confidence Information Ontology","Cell Ontology","Collembola Anatomy Ontology","CLO: Cell Line Ontology","Clytia hemisphaerica Development and Anatomy Ontology (CLYH)","Clinical measurement ontology","Core Ontology for Biology and Biomedicine","Coleoptera Anatomy Ontology","Contributor Role Ontology","Ctenophore Ontology","Dicty Anatomy Ontology (DDANAT)","Dicty Phenotype Ontology (DDPHENO)","Drug-drug Interaction and Drug-drug Interaction Evidence Ontology","Disease Drivers","Human Disease Ontology","Drosophila Phenotype Ontology (DPO)","The Drug Ontology","Data Use Ontology","Echinoderm Anatomy and Development Ontology","Evidence &amp; Conclusion Ontology (ECO)","An ontology of core ecological entities","Environment Exposure Ontology","Mouse Developmental Anatomy Ontology","The Environment Ontology","Epilepsy Ontology (EPIO)","Exposure ontology (ExO)","Fungal gross anatomy","Drosophila Anatomy Ontology (DAO)","FlyBase Controlled Vocabulary (FBcv)","Drosophila Developmental Ontology","Food Interactions with Drugs Evidence Ontology","Flora Phenotype Ontology","Food-Biomarker Ontology","Food Ontology","FuTRES Ontology of Vertebrate Traits","Fission Yeast Phenotype Ontology (FYPO)","Genomics Cohorts Knowledge Ontology","Genomic Epidemiology Ontology","GENO ontology","Geographical Entity Ontology","Glycan Naming Ontology","Gene Ontology","Human Ancestry Ontology","Hymenoptera Anatomy Ontology","Homology Ontology","Human Developmental Stages","Health Surveillance Ontology","Hypertension Ontology For Clinical Data","Information Artifact Ontology (IAO)","ICEO: Ontology of Integrative and Conjugative Elements","Informed Consent Ontology (ICO)","Infectious Disease Ontology","INO: Interaction Network Ontology","clinical LABoratory Ontology","Lepidoptera Anatomy Ontology","Mouse adult gross anatomy","Medical Action Ontology","Microbial Conditions Ontology","Model Card Report Ontology","Mental Functioning Ontology","Emotion Ontology","Mental Disease Ontology","Molecular Interactions Controlled Vocabulary","Minimum Information for A Phylogenetic Analysis (MIAPA) Ontology","Measurement method ontology","Mouse Developmental Stages","Protein modification","Mondo Disease Ontology","MOP","The Mammalian Phenotype Ontology","Mouse pathology ontology","Minimum PDDI Information Ontology","MHC Restriction Ontology","Mass spectrometry ontology","Neuro Behavior Ontology","NCBI organismal classification","NCI Thesaurus OBO Edition","Non-Coding RNA Ontology"," Next generation biobanking ontology(NGBO).","NOMEN - A nomenclatural ontology for biological names","OAE: Ontology of Adverse Events","Ontology of Arthropod Circulatory Systems","Ontology of Biological Attributes (OBA)","OBCS: Ontology of Biological and Clinical Statistics","Ontology for Biomedical Investigations","Ontology for BIoBanking (OBIB)","Occupation Ontology (OccO)","OGG: Ontology of Genes and Genomes","Ontology for General Medical Science","Oral Health and Disease Ontology","OHMI: Ontology of Host-Microbiome Interactions","OHPI: Ontology of Host-Pathogen Interactions","Ontology for MIRNA Target","OBO Metadata Ontology","Ontology of Microbial Phenotypes","Ontology for Modeling and Representation of Social Entities","Ontology for Nutritional Epidemiology","Ontology for Nutritional Studies","OntoAvida: ontology for Avida digital evolution platform.","Obstetric and Neonatal Ontology","Ontology of Organizational Structures of Trauma centers and Trauma Systems","Ontology for Parasite Lifecycle","OPMI: Ontology of Precision Medicine and Investigation","Ontology for RNA sequencing (ORNASEQ)","OVAE: Ontology of Vaccine Adverse Events","PATO - the Phenotype And Trait Ontology","Provisional Cell Ontology","Population and Community Ontology","The Prescription of Drugs Ontology","Pathogen Host Interactions Phenotype Ontology","Planarian Anatomy Ontology (PLANA)","Planarian Phenotype Ontology (PLANP)","Plant Ontology","Porifera (sponge) ontology","Plant Phenology Ontology","PRotein Ontology (PRO)","PROCO: PROcess Chemistry Ontology","Performance Summary Display Ontology","Plant Stress Ontology","Pathway ontology","Radiation Biology Ontology","Relation Ontology","Rat Strain Ontology","RXNO","Ontology for Scientific Evidence and Provenance Information","Sequence types and features ontology","Spider Ontology","STATO: the statistical methods ontology","Software ontology","Symptom Ontology","terms4FAIRskills","Taxonomic rank vocabulary","Plant Trait Ontology","Pathogen Transmission Ontology","Teleost taxonomy ontology","TOXic Process Ontology (TXPO)","Uber-anatomy ontology","Units of measurement ontology","Vertebrate Breed Ontology","Vaccine Ontology","The Vertebrate Trait Ontology","Vertebrate Taxonomy Ontology","C. elegans Gross Anatomy Ontology","C. elegans Development Ontology","C elegans Phenotype Ontology","Xenopus Anatomy Ontology","Experimental condition ontology","HUPO-PSI cross-linking and derivatization reagents controlled vocabulary","Xenopus Phenotype Ontology","Zebrafish Experimental Conditions Ontology","Zebrafish Anatomy Ontology (ZFA)","Zebrafish developmental stages ontology","Zebrafish Phenotype Ontology (ZP)","GSSO - the Gender, Sex, and Sexual Orientation ontology","Human Phenotype Ontology","Kinetic Simulation Algorithm Ontology","Systems Biology Ontology","Sickle Cell Disease Ontology","Physico-chemical methods and properties","Mathematical Modelling Ontology","Physico-chemical process","Social Insect Behavior Ontology","VEuPathDB Ontology","Biological Imaging Methods Ontology","Mammalian Feeding Muscle Ontology","Ontology of Prokaryotic Phenotypic and Metabolic Characters","Ontology of Genetic Susceptibility Factor","Anatomical Entity Ontology","Cephalopod Ontology","Human developmental anatomy, abstract","Foundational Model of Anatomy Ontology (subset)","Gazetteer","Malaria Ontology","Mosquito insecticide resistance","Tick Anatomy Ontology","Mosquito gross anatomy ontology","Unipathway","Adverse Event Reporting Ontology","Beta Cell Genomics Ontology","Bilateria anatomy","Mouse gross anatomy and development, timed","Fly taxonomy","Ontologized MIABIS","Teleost Anatomy Ontology","Vertebrate Skeletal Anatomy Ontology-","Semantic Mapping Vocabulary","SwissLipids","LIPID MAPS","","Chemical Analysis Metadata Platform (ChAMP) Ontology version 0.2","The RDF Schema vocabulary (RDFS)","The OWL 2 Schema vocabulary (OWL 2)","","","","","","Metabolomics Standards Initiative Ontology (MSIO)","DICOM Controlled Terminology","Sustainable Development Goals Interface Ontology","Ensembl Glossary","Experimental Factor Ontology","Orphanet Rare Disease Ontology","Terminology for Description of Dynamics","Bioinformatics operations, data types, formats, identifiers and topics","Cell Cycle Ontology","Regulation of Gene Expression Ontology","Regulation of Transcription Ontology","Gene Expression Ontology","Cellular Microscopy Phenotype Ontology","eNanoMapper ontology","Probability Distribution Ontology","PRIDE Controlled Vocabulary","BioAssay Ontology","Semanticscience Integrated Ontology","Unimod protein modification database for mass spectrometry","nuclear magnetic resonance CV","Allotrope Merged Ontology Suite","FAIRsharing Subject Ontology (SRAO)","Human Cell Atlas Ontology","The COVID-19 Infectious Disease Ontology","The Behaviour Change Intervention Ontology","CoVoc Coronavirus Vocabulary","Ontology of units of Measure (OM)","REPRODUCE-ME Ontology","Provenance Ontology (PROV-O)","Orthology Ontology","Cryo Electron Microscopy ontology","SNOMED CT (International Edition)","ShareLoc","Human Reference Atlas Common Coordinate Framework Ontology","Critical Path Ontology"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Prefix<\/th>\n      <th>Namespace<\/th>\n      <th>Title<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Prefix","targets":1},{"name":"Namespace","targets":2},{"name":"Title","targets":3}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div class="section level2">
<h2 id="a-brief-rols-overview">A Brief rols overview<a class="anchor" aria-label="anchor" href="#a-brief-rols-overview"></a>
</h2>
<p>The <em><a href="https://bioconductor.org/packages/3.19/rols" class="external-link">rols</a></em> package
is build around a few classes that enable to query the OLS and retrieve,
store and manipulate data. Each of these classes are described in more
details in their respective manual pages. We start by loading the
package.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="http://lgatto.github.io/rols/" class="external-link">"rols"</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="ontologies">Ontologies<a class="anchor" aria-label="anchor" href="#ontologies"></a>
</h3>
<p>The <code>Ontology</code> and <code>Ontologies</code> classes can
store information about single of multiple ontologies. The latter can be
easily subset using <code>[</code> and <code>[[</code>, as one would for
lists.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ol</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Ontologies.html">Ontologies</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## ⠙ Iterating 7 done (3.4/s) | 2s</span></span></code></pre>
<pre><code><span><span class="co">## ⠙ Iterating 13 done (4.3/s) | 3s</span></span></code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ol</span></span></code></pre></div>
<pre><code><span><span class="co">## Object of class 'Ontologies' with 251 entries</span></span>
<span><span class="co">##    ADO, AGRO ... CCF, CPONT</span></span></code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="../reference/Ontologies.html">olsNamespace</a></span><span class="op">(</span><span class="va">ol</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "ado"       "agro"      "aism"      "amphx"     "apo"       "apollo_sv"</span></span></code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ol</span><span class="op">[[</span><span class="st">"bspo"</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## Ontology: Biological Spatial Ontology (bspo)  </span></span>
<span><span class="co">##   An ontology for respresenting spatial concepts, anatomical axes,</span></span>
<span><span class="co">##   gradients, regions, planes, sides and surfaces. These concepts can be</span></span>
<span><span class="co">##   used at multiple biological scales and in a diversity of taxa,</span></span>
<span><span class="co">##   including plants, animals and fungi. The BSPO is used to provide a</span></span>
<span><span class="co">##   source of anatomical location descriptors for logically defining</span></span>
<span><span class="co">##   anatomical entity classes in anatomy ontologies.</span></span>
<span><span class="co">##    Loaded: 2024-02-09 Updated: 2024-02-09 Version: 2023-05-27 </span></span>
<span><span class="co">##    169 terms  236 properties  18 individuals</span></span></code></pre>
<p>It is also possible to initialise a single ontology</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bspo</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Ontologies.html">Ontology</a></span><span class="op">(</span><span class="st">"bspo"</span><span class="op">)</span></span>
<span><span class="va">bspo</span></span></code></pre></div>
<pre><code><span><span class="co">## Ontology: Biological Spatial Ontology (bspo)  </span></span>
<span><span class="co">##   An ontology for respresenting spatial concepts, anatomical axes,</span></span>
<span><span class="co">##   gradients, regions, planes, sides and surfaces. These concepts can be</span></span>
<span><span class="co">##   used at multiple biological scales and in a diversity of taxa,</span></span>
<span><span class="co">##   including plants, animals and fungi. The BSPO is used to provide a</span></span>
<span><span class="co">##   source of anatomical location descriptors for logically defining</span></span>
<span><span class="co">##   anatomical entity classes in anatomy ontologies.</span></span>
<span><span class="co">##    Loaded: 2024-02-09 Updated: 2024-02-09 Version: 2023-05-27 </span></span>
<span><span class="co">##    169 terms  236 properties  18 individuals</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="terms">Terms<a class="anchor" aria-label="anchor" href="#terms"></a>
</h3>
<p>Single ontology terms are stored in <code>Term</code> objects. When
more terms need to be manipulated, they are stored as <code>Terms</code>
objects. It is easy to obtain all terms of an ontology of interest, and
the resulting <code>Terms</code> object can be subset using
<code>[</code> and <code>[[</code>, as one would for lists.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bspotrms</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Terms.html">Terms</a></span><span class="op">(</span><span class="va">bspo</span><span class="op">)</span> <span class="co">## or Terms("bspo")</span></span>
<span><span class="va">bspotrms</span></span></code></pre></div>
<pre><code><span><span class="co">## Object of class 'Terms' with 169 entries</span></span>
<span><span class="co">##  From the BSPO ontology</span></span>
<span><span class="co">##   BFO:0000002, BFO:0000003 ... IAO:0000409, PATO:0000001</span></span></code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bspotrms</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## Object of class 'Terms' with 10 entries</span></span>
<span><span class="co">##  From the BSPO ontology</span></span>
<span><span class="co">##   BFO:0000002, BFO:0000003 ... BFO:0000023, BFO:0000031</span></span></code></pre>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bspotrms</span><span class="op">[[</span><span class="st">"BSPO:0000092"</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## A Term from the BSPO ontology: BSPO:0000092 </span></span>
<span><span class="co">##  Label: anatomical compartment boundary</span></span>
<span><span class="co">##   to be merged into CARO</span></span></code></pre>
<p>It is also possible to initialise a single term</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">trm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Terms.html">Term</a></span><span class="op">(</span><span class="va">bspo</span>, <span class="st">"BSPO:0000092"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/Terms.html">termId</a></span><span class="op">(</span><span class="va">trm</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "BSPO:0000092"</span></span></code></pre>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/Terms.html">termLabel</a></span><span class="op">(</span><span class="va">trm</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "anatomical compartment boundary"</span></span></code></pre>
<p>It is then possible to extract the <code>ancestors</code>,
<code>descendants</code>, <code>parents</code> and <code>children</code>
terms. Each of these functions return a <code>Terms</code> object</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/Terms.html">parents</a></span><span class="op">(</span><span class="va">trm</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Object of class 'Terms' with 1 entries</span></span>
<span><span class="co">##  From the BSPO ontology</span></span>
<span><span class="co">## CARO:0000010</span></span></code></pre>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/Terms.html">children</a></span><span class="op">(</span><span class="va">trm</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Object of class 'Terms' with 6 entries</span></span>
<span><span class="co">##  From the BSPO ontology</span></span>
<span><span class="co">##   BSPO:0000094, BSPO:0000093 ... BSPO:0000041, BSPO:0000040</span></span></code></pre>
<p>Finally, a single term or terms object can be coerced to a
<code>data.frame</code> using <code>as(x, "data.frame")</code>.</p>
</div>
<div class="section level3">
<h3 id="properties">Properties<a class="anchor" aria-label="anchor" href="#properties"></a>
</h3>
<p>Properties (relationships) of single or multiple terms or complete
ontologies can be queries with the <code>properties</code> method, as
briefly illustrated below.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">trm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Terms.html">Term</a></span><span class="op">(</span><span class="st">"uberon"</span>, <span class="st">"UBERON:0002107"</span><span class="op">)</span></span>
<span><span class="va">trm</span></span></code></pre></div>
<pre><code><span><span class="co">## A Term from the UBERON ontology: UBERON:0002107 </span></span>
<span><span class="co">##  Label: liver</span></span>
<span><span class="co">##   An exocrine gland which secretes bile and functions in metabolism of</span></span>
<span><span class="co">##   protein and carbohydrate and fat, synthesizes substances involved in</span></span>
<span><span class="co">##   the clotting of the blood, synthesizes vitamin A, detoxifies poisonous</span></span>
<span><span class="co">##   substances, stores glycogen, and breaks down worn-out erythrocytes[GO].</span></span></code></pre>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Properties.html">Properties</a></span><span class="op">(</span><span class="va">trm</span><span class="op">)</span></span>
<span><span class="va">p</span></span></code></pre></div>
<pre><code><span><span class="co">## Object of class 'Properties' with 158 entries</span></span>
<span><span class="co">##  From the UBERON ontology</span></span>
<span><span class="co">##   hepatobiliary system, exocrine system ... liver serosa, liver subserosa</span></span></code></pre>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## A Property from the UBERON ontology: UBERON:0002423 </span></span>
<span><span class="co">##  Label: hepatobiliary system</span></span></code></pre>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/Terms.html">termLabel</a></span><span class="op">(</span><span class="va">p</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "hepatobiliary system"</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="use-case">Use case<a class="anchor" aria-label="anchor" href="#use-case"></a>
</h2>
<p>A researcher might be interested in the trans-Golgi network.
Searching the OLS is assured by the <code>OlsSearch</code> and
<code>olsSearch</code> classes/functions. The first step is to defined
the search query with <code>OlsSearch</code>, as shown below. This
creates an search object of class <code>OlsSearch</code> that stores the
query and its parameters. In records the number of requested results
(default is 20) and the total number of possible results (there are
16844 results across all ontologies, in this case). At this stage, the
results have not yet been downloaded, as shown by the 0 responses.</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/OlsSearch.html">OlsSearch</a></span><span class="op">(</span>q <span class="op">=</span> <span class="st">"trans-golgi network"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Object of class 'OlsSearch':</span></span>
<span><span class="co">##   query: trans-golgi network </span></span>
<span><span class="co">##   requested: 20 (out of 16844)</span></span>
<span><span class="co">##   response(s): 0</span></span></code></pre>
<p>16844 results are probably too many to be relevant. Below we show how
to perform an exact search by setting <code>exact = TRUE</code>, and
limiting the search the the GO ontology by specifying
<code>ontology = "GO"</code>, or doing both.</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/OlsSearch.html">OlsSearch</a></span><span class="op">(</span>q <span class="op">=</span> <span class="st">"trans-golgi network"</span>, exact <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Object of class 'OlsSearch':</span></span>
<span><span class="co">##   query: trans-golgi network </span></span>
<span><span class="co">##   requested: 20 (out of 215)</span></span>
<span><span class="co">##   response(s): 0</span></span></code></pre>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/OlsSearch.html">OlsSearch</a></span><span class="op">(</span>q <span class="op">=</span> <span class="st">"trans-golgi network"</span>, ontology <span class="op">=</span> <span class="st">"GO"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Object of class 'OlsSearch':</span></span>
<span><span class="co">##   ontolgy: GO </span></span>
<span><span class="co">##   query: trans-golgi network </span></span>
<span><span class="co">##   requested: 20 (out of 1109)</span></span>
<span><span class="co">##   response(s): 0</span></span></code></pre>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/OlsSearch.html">OlsSearch</a></span><span class="op">(</span>q <span class="op">=</span> <span class="st">"trans-golgi network"</span>, ontology <span class="op">=</span> <span class="st">"GO"</span>, exact <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Object of class 'OlsSearch':</span></span>
<span><span class="co">##   ontolgy: GO </span></span>
<span><span class="co">##   query: trans-golgi network </span></span>
<span><span class="co">##   requested: 20 (out of 25)</span></span>
<span><span class="co">##   response(s): 0</span></span></code></pre>
<p>One case set the <code>rows</code> argument to set the number of
desired results.</p>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/OlsSearch.html">OlsSearch</a></span><span class="op">(</span>q <span class="op">=</span> <span class="st">"trans-golgi network"</span>, ontology <span class="op">=</span> <span class="st">"GO"</span>, rows <span class="op">=</span> <span class="fl">200</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Object of class 'OlsSearch':</span></span>
<span><span class="co">##   ontolgy: GO </span></span>
<span><span class="co">##   query: trans-golgi network </span></span>
<span><span class="co">##   requested: 200 (out of 1109)</span></span>
<span><span class="co">##   response(s): 0</span></span></code></pre>
<p>See <code><a href="../reference/OlsSearch.html">?OlsSearch</a></code> for details about retrieving many
results.</p>
<p>Let’s proceed with the exact search and retrieve the results. Even if
we request the default 20 results, only the 215 relevant result will be
retrieved. The <code>olsSearch</code> function updates the previously
created object (called <code>qry</code> below) by adding the results to
it.</p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">qry</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/OlsSearch.html">OlsSearch</a></span><span class="op">(</span>q <span class="op">=</span> <span class="st">"trans-golgi network"</span>, exact <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="op">(</span><span class="va">qry</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/OlsSearch.html">olsSearch</a></span><span class="op">(</span><span class="va">qry</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Object of class 'OlsSearch':</span></span>
<span><span class="co">##   query: trans-golgi network </span></span>
<span><span class="co">##   requested: 20 (out of 215)</span></span>
<span><span class="co">##   response(s): 20</span></span></code></pre>
<p>We can now transform this search result object into a fully fledged
<code>Terms</code> object or a <code>data.frame</code>.</p>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span><span class="va">qtrms</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/methods/as.html" class="external-link">as</a></span><span class="op">(</span><span class="va">qry</span>, <span class="st">"Terms"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Object of class 'Terms' with 20 entries</span></span>
<span><span class="co">##  From the NCIT, PR, GO, ZP, PW ontologies</span></span>
<span><span class="co">##   NCIT:C33802, PR:O43493 ... GO:0042147, PW:0000426</span></span></code></pre>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">qdrf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/methods/as.html" class="external-link">as</a></span><span class="op">(</span><span class="va">qry</span>, <span class="st">"data.frame"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 'data.frame':    20 obs. of  8 variables:</span></span>
<span><span class="co">##  $ iri            : chr  "http://purl.obolibrary.org/obo/NCIT_C33802" "http://purl.obolibrary.org/obo/PR_O43493" "http://purl.obolibrary.org/obo/GO_0005802" "http://purl.obolibrary.org/obo/GO_0032588" ...</span></span>
<span><span class="co">##  $ ontology_name  : chr  "ncit" "pr" "go" "go" ...</span></span>
<span><span class="co">##  $ ontology_prefix: chr  "NCIT" "PR" "GO" "GO" ...</span></span>
<span><span class="co">##  $ short_form     : chr  "NCIT_C33802" "PR_O43493" "GO_0005802" "GO_0032588" ...</span></span>
<span><span class="co">##  $ description    :List of 20</span></span>
<span><span class="co">##   ..$ : chr "A network of membrane components where vesicles bud off the Golgi apparatus to bring proteins, membranes and ot"| __truncated__</span></span>
<span><span class="co">##   ..$ : chr  "A trans-Golgi network integral membrane protein 2 that is encoded in the genome of human." "Category=organism-gene."</span></span>
<span><span class="co">##   ..$ : chr  "The network of interconnected tubular and cisternal structures located within the Golgi apparatus on the side d"| __truncated__ "There are different opinions about whether the TGN should be considered part of the Golgi apparatus or not. We "| __truncated__</span></span>
<span><span class="co">##   ..$ : chr "The lipid bilayer surrounding any of the compartments that make up the trans-Golgi network."</span></span>
<span><span class="co">##   ..$ : chr "Abnormal(ly) mislocalised (of) enterocyte of trans-Golgi network."</span></span>
<span><span class="co">##   ..$ : chr "A vesicle that mediates transport between the trans-Golgi network and other parts of the cell."</span></span>
<span><span class="co">##   ..$ : chr "The side (leaflet) of the trans-Golgi network transport vesicle membrane that faces the lumen."</span></span>
<span><span class="co">##   ..$ : chr "The side (leaflet) of the trans-Golgi network transport vesicle membrane that faces the cytoplasm."</span></span>
<span><span class="co">##   ..$ : chr "The lipid bilayer surrounding a vesicle transporting substances between the trans-Golgi network and other parts of the cell."</span></span>
<span><span class="co">##   ..$ : chr "The volume enclosed within the membrane of a trans-Golgi network transport vesicle."</span></span>
<span><span class="co">##   ..$ : chr "A clathrin coat found on a vesicle of the trans-Golgi network."</span></span>
<span><span class="co">##   ..$ : chr  "A trans-Golgi network integral membrane protein 1 that is encoded in the genome of rat." "Category=organism-gene."</span></span>
<span><span class="co">##   ..$ : chr "A process which results in the assembly, arrangement of constituent parts, or disassembly of a trans-Golgi network membrane."</span></span>
<span><span class="co">##   ..$ : chr  "A trans-Golgi network integral membrane protein 1 that is encoded in the genome of mouse." "Category=organism-gene."</span></span>
<span><span class="co">##   ..$ : chr "The directed movement of substances, in membrane-bounded vesicles, from the trans-Golgi network to the recycling endosomes."</span></span>
<span><span class="co">##   ..$ : chr "The directed movement of substances from the plasma membrane back to the trans-Golgi network, mediated by vesicles."</span></span>
<span><span class="co">##   ..$ : chr "The directed movement of proteins from the Golgi to the plasma membrane in transport vesicles that move from th"| __truncated__</span></span>
<span><span class="co">##   ..$ : chr "The directed movement of substances from the vacuole to the trans-Golgi network; this occurs in yeast via the p"| __truncated__</span></span>
<span><span class="co">##   ..$ : chr "The directed movement of membrane-bounded vesicles from endosomes back to the trans-Golgi network where they ar"| __truncated__</span></span>
<span><span class="co">##   ..$ : chr "In the secretory pathway, protein sorting, mainly in trans-Golgi Network (TGN), but also in other compartments,"| __truncated__</span></span>
<span><span class="co">##  $ label          : chr  "Trans-Golgi Network" "trans-Golgi network integral membrane protein 2 (human)" "trans-Golgi network" "trans-Golgi network membrane" ...</span></span>
<span><span class="co">##  $ obo_id         : chr  "NCIT:C33802" "PR:O43493" "GO:0005802" "GO:0032588" ...</span></span>
<span><span class="co">##  $ type           : chr  "class" "class" "class" "class" ...</span></span></code></pre>
<p>In this case, we can see that we actually retrieve the same term used
across different ontologies. In such cases, it might be useful to keep
only non-redundant term instances. Here, this would have been equivalent
to searching the ncit, pr, go, go, zp, go, go, go, go, go, go, pr, go,
pr, go, go, go, go, go, pw ontology</p>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">qtrms</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">qtrms</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/Terms.html">termOntology</a></span><span class="op">(</span><span class="va">qtrms</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## NCIT:C33802   PR:O43493  GO:0005802  GO:0032588  ZP:0142408  GO:0030140 </span></span>
<span><span class="co">##      "ncit"        "pr"        "go"        "go"        "zp"        "go" </span></span>
<span><span class="co">##  GO:0098540  GO:0098541  GO:0012510  GO:0098564  GO:0030130   PR:P19814 </span></span>
<span><span class="co">##        "go"        "go"        "go"        "go"        "go"        "pr" </span></span>
<span><span class="co">##  GO:0098629   PR:Q62313  GO:0044795  GO:0035526  GO:0043001  GO:0045018 </span></span>
<span><span class="co">##        "go"        "pr"        "go"        "go"        "go"        "go" </span></span>
<span><span class="co">##  GO:0042147  PW:0000426 </span></span>
<span><span class="co">##        "go"        "pw"</span></span></code></pre>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/Terms.html">termNamespace</a></span><span class="op">(</span><span class="va">qtrms</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## $`NCIT:C33802`</span></span>
<span><span class="co">## NULL</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $`PR:O43493`</span></span>
<span><span class="co">## [1] "protein"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $`GO:0005802`</span></span>
<span><span class="co">## [1] "cellular_component"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $`GO:0032588`</span></span>
<span><span class="co">## [1] "cellular_component"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $`ZP:0142408`</span></span>
<span><span class="co">## NULL</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $`GO:0030140`</span></span>
<span><span class="co">## [1] "cellular_component"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $`GO:0098540`</span></span>
<span><span class="co">## [1] "cellular_component"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $`GO:0098541`</span></span>
<span><span class="co">## [1] "cellular_component"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $`GO:0012510`</span></span>
<span><span class="co">## [1] "cellular_component"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $`GO:0098564`</span></span>
<span><span class="co">## [1] "cellular_component"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $`GO:0030130`</span></span>
<span><span class="co">## [1] "cellular_component"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $`PR:P19814`</span></span>
<span><span class="co">## [1] "protein"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $`GO:0098629`</span></span>
<span><span class="co">## [1] "biological_process"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $`PR:Q62313`</span></span>
<span><span class="co">## [1] "protein"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $`GO:0044795`</span></span>
<span><span class="co">## [1] "biological_process"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $`GO:0035526`</span></span>
<span><span class="co">## [1] "biological_process"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $`GO:0043001`</span></span>
<span><span class="co">## [1] "biological_process"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $`GO:0045018`</span></span>
<span><span class="co">## [1] "biological_process"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $`GO:0042147`</span></span>
<span><span class="co">## [1] "biological_process"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $`PW:0000426`</span></span>
<span><span class="co">## [1] "pathway"</span></span></code></pre>
<p>Below, we execute the same query using the <em><a href="https://bioconductor.org/packages/3.19/GO.db" class="external-link">GO.db</a></em>
package.</p>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"GO.db"</span><span class="op">)</span></span>
<span><span class="va">GOTERM</span><span class="op">[[</span><span class="st">"GO:0005802"</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## GOID: GO:0005802</span></span>
<span><span class="co">## Term: trans-Golgi network</span></span>
<span><span class="co">## Ontology: CC</span></span>
<span><span class="co">## Definition: The network of interconnected tubular and cisternal</span></span>
<span><span class="co">##     structures located within the Golgi apparatus on the side distal to</span></span>
<span><span class="co">##     the endoplasmic reticulum, from which secretory vesicles emerge.</span></span>
<span><span class="co">##     The trans-Golgi network is important in the later stages of protein</span></span>
<span><span class="co">##     secretion where it is thought to play a key role in the sorting and</span></span>
<span><span class="co">##     targeting of secreted proteins to the correct destination.</span></span>
<span><span class="co">## Synonym: TGN</span></span>
<span><span class="co">## Synonym: trans Golgi network</span></span>
<span><span class="co">## Synonym: Golgi trans face</span></span>
<span><span class="co">## Synonym: Golgi trans-face</span></span>
<span><span class="co">## Synonym: late Golgi</span></span>
<span><span class="co">## Synonym: maturing face</span></span>
<span><span class="co">## Synonym: trans face</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="on-line-vs--off-line-data">On-line vs. off-line data<a class="anchor" aria-label="anchor" href="#on-line-vs--off-line-data"></a>
</h2>
<p>It is possible to observe different results with <em><a href="https://bioconductor.org/packages/3.19/rols" class="external-link">rols</a></em> and
<em><a href="https://bioconductor.org/packages/3.19/GO.db" class="external-link">GO.db</a></em>, as a
result of the different ways they access the data. <em><a href="https://bioconductor.org/packages/3.19/rols" class="external-link">rols</a></em> or
<em><a href="https://bioconductor.org/packages/3.19/biomaRt" class="external-link">biomaRt</a></em>
perform direct online queries, while <em><a href="https://bioconductor.org/packages/3.19/GO.db" class="external-link">GO.db</a></em> and
other annotation packages use database snapshot that are updated every
release.</p>
<p>Both approaches have advantages. While online queries allow to obtain
the latest up-to-date information, such approaches rely on network
availability and quality. If reproducibility is a major issue, the
version of the database to be queried can easily be controlled with
off-line approaches. In the case of <em><a href="https://bioconductor.org/packages/3.19/rols" class="external-link">rols</a></em>,
although the load date of a specific ontology can be queried with
<code>olsVersion</code>, it is not possible to query a specific version
of an ontology.</p>
</div>
<div class="section level2">
<h2 id="changes">Changes<a class="anchor" aria-label="anchor" href="#changes"></a>
</h2>
<div class="section level3">
<h3 id="version-2-0">Version 2.0<a class="anchor" aria-label="anchor" href="#version-2-0"></a>
</h3>
<p><em><a href="https://bioconductor.org/packages/3.19/rols" class="external-link">rols</a></em> 2.0 has
substantially changed. While the table below shows some correspondence
between the old and new interface, this is not always the case. The new
interface relies on the <code>Ontology</code>/<code>Ontologies</code>,
<code>Term</code>/<code>Terms</code> and <code>OlsSearch</code> classes,
that need to be instantiated and can then be queried, as described
above.</p>
<table class="table">
<thead><tr class="header">
<th>version &lt; 1.99</th>
<th>version &gt;= 1.99</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><code>ontologyLoadDate</code></td>
<td>
<code>olsLoaded</code> and <code>olsUpdated</code>
</td>
</tr>
<tr class="even">
<td><code>ontologyNames</code></td>
<td><code>Ontologies</code></td>
</tr>
<tr class="odd">
<td><code>olsVersion</code></td>
<td><code>olsVersion</code></td>
</tr>
<tr class="even">
<td><code>allIds</code></td>
<td><code>terms</code></td>
</tr>
<tr class="odd">
<td><code>isIdObsolete</code></td>
<td><code>isObsolete</code></td>
</tr>
<tr class="even">
<td><code>rootId</code></td>
<td><code>olsRoot</code></td>
</tr>
<tr class="odd">
<td><code>olsQuery</code></td>
<td>
<code>OlsSearch</code> and <code>olsSearch</code>
</td>
</tr>
</tbody>
</table>
<p>Not all functionality is currently available. If there is anything
that you need but not available in the new version, please contact the
maintained by opening an <a href="https://github.com/lgatto/rols/issues" class="external-link">issue</a> on the package
development site.</p>
</div>
</div>
<div class="section level2">
<h2 id="version-2-99">Version 2.99<a class="anchor" aria-label="anchor" href="#version-2-99"></a>
</h2>
<ul>
<li>
<code>rols</code> version &gt;= 2.99 has been refactored to use the
OLS4 REST API.</li>
<li>REST queries now use <a href="https://httr2.r-lib.org/" class="external-link">httr2</a>
instead of superseded <code>httr</code>.</li>
<li>The term(s) constructors are capitalised as <code><a href="../reference/Terms.html">Term()</a></code> and
<code><a href="../reference/Terms.html">Terms()</a></code>.</li>
<li>The properties constructor is capitalised as
<code><a href="../reference/Properties.html">Properties()</a></code>.</li>
<li>Some class definitions have been updated to accomodate changes in
the data received by OLS. Some function have been dropped.</li>
</ul>
</div>
<div class="section level2">
<h2 id="cvparams">CVParams<a class="anchor" aria-label="anchor" href="#cvparams"></a>
</h2>
<p>The <code>CVParam</code> class is used to handle controlled
vocabulary. It can be used for user-defined parameters</p>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/CVParam.html">CVParam</a></span><span class="op">(</span>name <span class="op">=</span> <span class="st">"A user param"</span>, value <span class="op">=</span> <span class="st">"the value"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [, , A user param, the value]</span></span></code></pre>
<p>or official controlled vocabulary (which triggers a query to the OLS
service)</p>
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/CVParam.html">CVParam</a></span><span class="op">(</span>label <span class="op">=</span> <span class="st">"MS"</span>, accession <span class="op">=</span> <span class="st">"MS:1000073"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [MS, MS:1000073, electrospray ionization, ]</span></span></code></pre>
<p>See <code><a href="../reference/CVParam.html">?CVParam</a></code> for more details and examples.</p>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<pre><code><span><span class="co">## R Under development (unstable) (2024-02-07 r85873)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.3 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] DT_0.31              rols_2.99.0          GO.db_3.18.0        </span></span>
<span><span class="co">## [4] AnnotationDbi_1.65.2 IRanges_2.37.1       S4Vectors_0.41.3    </span></span>
<span><span class="co">## [7] Biobase_2.63.0       BiocGenerics_0.49.1  BiocStyle_2.31.0    </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] KEGGREST_1.43.0         xfun_0.42               bslib_0.6.1            </span></span>
<span><span class="co">##  [4] httr2_1.0.0             htmlwidgets_1.6.4       vctrs_0.6.5            </span></span>
<span><span class="co">##  [7] tools_4.4.0             crosstalk_1.2.1         bitops_1.0-7           </span></span>
<span><span class="co">## [10] curl_5.2.0              RSQLite_2.3.5           blob_1.2.4             </span></span>
<span><span class="co">## [13] pkgconfig_2.0.3         desc_1.4.3              lifecycle_1.0.4        </span></span>
<span><span class="co">## [16] GenomeInfoDbData_1.2.11 compiler_4.4.0          stringr_1.5.1          </span></span>
<span><span class="co">## [19] textshaping_0.3.7       Biostrings_2.71.2       GenomeInfoDb_1.39.6    </span></span>
<span><span class="co">## [22] htmltools_0.5.7         sass_0.4.8              RCurl_1.98-1.14        </span></span>
<span><span class="co">## [25] yaml_2.3.8              pkgdown_2.0.7.9000      crayon_1.5.2           </span></span>
<span><span class="co">## [28] jquerylib_0.1.4         ellipsis_0.3.2          cachem_1.0.8           </span></span>
<span><span class="co">## [31] digest_0.6.34           stringi_1.8.3           purrr_1.0.2            </span></span>
<span><span class="co">## [34] bookdown_0.37           fastmap_1.1.1           cli_3.6.2              </span></span>
<span><span class="co">## [37] magrittr_2.0.3          rappdirs_0.3.3          bit64_4.0.5            </span></span>
<span><span class="co">## [40] rmarkdown_2.25          XVector_0.43.1          httr_1.4.7             </span></span>
<span><span class="co">## [43] bit_4.0.5               ragg_1.2.7              png_0.1-8              </span></span>
<span><span class="co">## [46] memoise_2.0.1           evaluate_0.23           knitr_1.45             </span></span>
<span><span class="co">## [49] rlang_1.1.3             glue_1.7.0              DBI_1.2.1              </span></span>
<span><span class="co">## [52] BiocManager_1.30.22     jsonlite_1.8.8          R6_2.5.1               </span></span>
<span><span class="co">## [55] systemfonts_1.0.5       fs_1.6.3                zlibbioc_1.49.0</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Laurent Gatto.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.9000.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
