<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>An R interface to the Ontology Lookup Service • rols</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="An R interface to the Ontology Lookup Service">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">rols</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">2.11.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/rols.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><script src="rols_files/htmlwidgets-1.3/htmlwidgets.js"></script><script src="rols_files/jquery-1.12.4/jquery.min.js"></script><link href="rols_files/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="rols_files/datatables-binding-0.5/datatables.js"></script><link href="rols_files/dt-core-1.10.16/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="rols_files/dt-core-1.10.16/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="rols_files/dt-core-1.10.16/js/jquery.dataTables.min.js"></script><link href="rols_files/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet">
<script src="rols_files/crosstalk-1.0.0/js/crosstalk.min.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>An R interface to the Ontology Lookup Service</h1>
                        <h4 class="author">Laurent Gatto</h4>
            <address class="author_afil">
      <a href="https://lgatto.github.io/about/" class="uri">https://lgatto.github.io/about/</a><br><div class="hidden name"><code>rols.Rmd</code></div>

    </address>
</div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      How to query the Ontology Lookup Service directly from R and how to create and parse controlled vocabulary.
    </div>
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p><em><a href="https://bioconductor.org/packages/3.9/rols">rols</a></em> is a Bioconductor package and should hence be installed using the dedicated functionality</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1">## try http:// if https:// URLs are not supported</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="cf">if</span> (<span class="op">!</span><span class="kw">requireNamespace</span>(<span class="st">"BiocManager"</span>, <span class="dt">quietly=</span><span class="ot">TRUE</span>))</a>
<a class="sourceLine" id="cb1-3" data-line-number="3">    <span class="kw">install.packages</span>(<span class="st">"BiocManager"</span>)</a>
<a class="sourceLine" id="cb1-4" data-line-number="4">BiocManager<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/BiocManager/topics/install">install</a></span>(<span class="st">"rols"</span>)</a></code></pre></div>
</div>
<div id="getting-help" class="section level2">
<h2 class="hasAnchor">
<a href="#getting-help" class="anchor"></a>Getting help</h2>
<p>To get help, either post your question on the <a href="https://support.bioconductor.org/">Bioconductor support site</a> or open an issue on the <em><a href="https://bioconductor.org/packages/3.9/rols">rols</a></em> <a href="https://github.com/lgatto/rols/issues">github page</a>.</p>
</div>
<div id="the-resource" class="section level2">
<h2 class="hasAnchor">
<a href="#the-resource" class="anchor"></a>The resource</h2>
<p>The <a href="http://www.ebi.ac.uk/ontology-lookup/">Ontology Lookup Service</a> (OLS) [<a href="http://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-7-97">1</a>, <a href="http://nar.oxfordjournals.org/content/36/suppl_2/W372">2</a>] is originally spin-off of the <a href="http://www.ebi.ac.uk/pride/archive/">PRoteomics IDEntifications</a> database (PRIDE) service, located at the EBI, and is now developed and maintained by the <a href="http://www.ebi.ac.uk/about/spot-team">Samples, Phenotypes and Ontologies</a> team at EMBL-EBI.</p>
</div>
<div id="the-package" class="section level2">
<h2 class="hasAnchor">
<a href="#the-package" class="anchor"></a>The package</h2>
<p>The OLS provides a REST interface to hundreds of ontologies from a single location with a unified output format. The <em><a href="https://bioconductor.org/packages/3.9/rols">rols</a></em> package make this possible from within R. Do do so, it relies on the <em><a href="https://CRAN.R-project.org/package=httr">httr</a></em> package to query the REST interface, and access and retrieve data.</p>
<p>There are 217 ontologies available in the OLS, listed in the table below. Their name is to be use to defined which ontology to query.</p>
<div id="htmlwidget-4fe227bfe20af78dd088" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-4fe227bfe20af78dd088">{"x":{"filter":"none","data":[["bco","bfo","bspo","bto","caro","cco","cdao","ceph","chebi","cheminf","chmo","cio","cl","clo","cmo","cmpo","co_320","co_321","co_322","co_323","co_324","co_325","co_327","co_330","co_331","co_333","co_334","co_335","co_336","co_337","co_338","co_339","co_340","co_341","co_343","co_345","co_346","co_347","co_348","co_350","co_356","co_357","co_359","co_360","cro","cteno","cvdo","ddanat","fma","gaz","genepio","geno","geo","gexo","go","hancestro","hao","hom","hp","hsapdv","iao","ico","ido","idomal","ino","ma","mf","mfoem","bao","aero","mfomd","mi","dinto","micro","bcgo","dron","miro","dpo","emap","edam","flopo","aeo","kisao","efo","mondo","agro","mop","lbo","ancestro","mpath","mamo","apo","nbo","nmrcv","aro","mfmo","atol","miapa","ddpheno","mp","mmusdv","dicom","dideo","mro","msio","doid","ncbitaxon","mpio","oarcs","duo","oae","oba","eco","obi","ogms","obib","peco","ogi","plana","ogsf","ncit","omp","ecocore","ppo","omrse","sepio","ontoneo","ncro","ehdaa2","sibo","emapa","oostt","sio","enm","obcs","ensemblglossary","ogg","spd","envo","opl","stato","eo","to","eol","ordo","trans","epo","ovae","pdro","ero","unimod","ohmi","po","eupath","vt","poro","olatdv","exo","vto","omiabis","reto","wbls","omit","fao","rexo","fbbi","wbphenotype","rnao","fbbt","phipo","rs","fbcv","xao","pride","rxno","fbdv","pw","xco","scdo","fix","xl","rex","flu","so","swo","symp","foodon","fypo","uo","tads","mirnao","upheno","taxrank","teddy","mmo","vario","tgma","mod","vo","zfa","wbbt","xlmod","ms","ohd","pato","pco","pdumdv","pr","probonto","ro","sbo","sdgio","sep","tto","uberon","zeco","zfs"],["BCO","BFO","BSPO","BTO","CARO","CCO","CDAO","CEPH","CHEBI","CHEMINF","CHMO","CIO","CL","CLO","CMO","CMPO","CO_320","CO_321","CO_322","CO_323","CO_324","CO_325","CO_327","CO_330","CO_331","CO_333","CO_334","CO_335","CO_336","CO_337","CO_338","CO_339","CO_340","CO_341","CO_343","CO_345","CO_346","CO_347","CO_348","CO_350","CO_356","CO_357","CO_359","CO_360","CRO","CTENO","CVDO","DDANAT","FMA","GAZ","GENEPIO","GENO","GEO","GeXO","GO","HANCESTRO","HAO","HOM","HP","HSAPDV","IAO","ICO","IDO","IDOMAL","INO","MA","MF","MFOEM","BAO","AERO","MFOMD","MI","DINTO","MICRO","BCGO","DRON","MIRO","FBcv","EMAP","EDAM","FLOPO","AEO","KISAO","EFO","MONDO","AGRO","MOP","LBO","ancestro","MPATH","MAMO","APO","NBO","NMR","ARO","MFMO","ATOL","MIAPA","DDPHENO","MP","MMUSDV","DICOM","DIDEO","MRO","MSIO","DOID","NCBITAXON","MPIO","OARCS","DUO","OAE","OBA","ECO","OBI","OGMS","OBIB","PECO","OGI","PLANA","OGSF","NCIT","OMP","ECOCORE","PPO","OMRSE","SEPIO","ONTONEO","NCRO","EHDAA2","SIBO","EMAPA","OOSTT","SIO","ENM","OBCS","ENSGLOSS","OGG","SPD","ENVO","OPL","STATO","EO","TO","EOL","ORDO","TRANS","EPO","OVAE","PDRO","ERO","UNIMOD","OHMI","PO","EUPATH","VT","PORO","OLATDV","EXO","VTO","OMIABIS","ReTO","WBLS","OMIT","FAO","ReXO","FBbi","WBPhenotype","RNAO","FBbt","PHIPO","RS","FBcv","XAO","PRIDE","RXNO","FBdv","PW","XCO","SCDO","FIX","XL","REX","FLU","SO","SWO","SYMP","FOODON","FYPO","UO","TADS","MIRNAO","UPHENO","TAXRANK","TEDDY","MMO","VARIO","TGMA","MOD","VO","ZFA","WBbt","XLMOD","MS","OHD","PATO","PCO","PDUMDV","PR","PROBONTO","RO","SBO","SDGIO","SEP","TTO","UBERON","ZECO","ZFS"],["bco","bfo","bspo","bto","caro","cco","cdao","ceph","chebi","cheminf","chmo","cio","cl","clo","cmo","cmpo","co_320","co_321","co_322","co_323","co_324","co_325","co_327","co_330","co_331","co_333","co_334","co_335","co_336","co_337","co_338","co_339","co_340","co_341","co_343","co_345","co_346","co_347","co_348","co_350","co_356","co_357","co_359","co_360","cro","cteno","cvdo","ddanat","fma","gaz","genepio","geno","geo","gexo","go","hancestro","hao","hom","hp","hsapdv","iao","ico","ido","idomal","ino","ma","mf","mfoem","bao","aero","mfomd","mi","dinto","micro","bcgo","dron","miro","dpo","emap","edam","flopo","aeo","kisao","efo","mondo","agro","mop","lbo","ancestro","mpath","mamo","apo","nbo","nmrcv","aro","mfmo","atol","miapa","ddpheno","mp","mmusdv","dicom","dideo","mro","msio","doid","ncbitaxon","mpio","oarcs","duo","oae","oba","eco","obi","ogms","obib","peco","ogi","plana","ogsf","ncit","omp","ecocore","ppo","omrse","sepio","ontoneo","ncro","ehdaa2","sibo","emapa","oostt","sio","enm","obcs","ensemblglossary","ogg","spd","envo","opl","stato","eo","to","eol","ordo","trans","epo","ovae","pdro","ero","unimod","ohmi","po","eupath","vt","poro","olatdv","exo","vto","omiabis","reto","wbls","omit","fao","rexo","fbbi","wbphenotype","rnao","fbbt","phipo","rs","fbcv","xao","pride","rxno","fbdv","pw","xco","scdo","fix","xl","rex","flu","so","swo","symp","foodon","fypo","uo","tads","mirnao","upheno","taxrank","teddy","mmo","vario","tgma","mod","vo","zfa","wbbt","xlmod","ms","ohd","pato","pco","pdumdv","pr","probonto","ro","sbo","sdgio","sep","tto","uberon","zeco","zfs"],["Biological Collections Ontology","Basic Formal Ontology","Biological Spatial Ontology","BRENDA tissue / enzyme source","Common Anatomy Reference Ontology","Cell Cycle Ontology","Comparative Data Analysis Ontology","Cephalopod Ontology","Chemical Entities of Biological Interest","chemical information ontology (cheminf) - information entities about chemical entities","Chemical Methods Ontology","Confidence Information Ontology","Cell Ontology","CLO: Cell Line Ontology","Clinical measurement ontology","Cellular Microscopy Phenotype Ontology","Rice ontology","Wheat ontology","Maize ontology","Barley ontology","Sorghum ontology","Banana ontology","Pearl millet ontology","Potato ontology","Sweet Potato ontology","Beet Ontology ontology","Cassava ontology","Common bean ontology","Soybean ontology","Groundnut ontology","Chickpea ontology","Lentil ontology","Cowpea ontology","Pigeonpea ontology","Yam ontology","Brachiaria ontology","Mungbean ontology","Castor bean ontology","Brassica ontology","Oat ontology","Vitis ontology","Woody Plant Ontology ontology","Sunflower ontology","Sugar Kelp trait ontology","Contributor Role Ontology","Ctenophore Ontology","Cardiovascular Disease Ontology","Dictyostelium discoideum anatomy","Foundational Model of Anatomy Ontology (subset)","Gazetteer","Genomic Epidemiology Ontology","Genotype Ontology","Geographical Entity Ontology","Gene Expression Ontology","Gene Ontology","Human Ancestry Ontology","Hymenoptera Anatomy Ontology","Homology Ontology","human phenotype ontology","Human Developmental Stages","Information Artifact Ontology","Informed Consent Ontology","Infectious Disease Ontology","Malaria Ontology","INO: Interaction Network Ontology","Mouse adult gross anatomy","Mental Functioning Ontology","Emotion Ontology","BioAssay Ontology","Adverse Event Reporting Ontology","Mental Disease Ontology","Molecular Interactions Controlled Vocabulary","The Drug-Drug Interactions Ontology","Ontology of Prokaryotic Phenotypic and Metabolic Characters","Beta Cell Genomics Ontology","The Drug Ontology","Mosquito insecticide resistance","Drosophila Phenotype Ontology","Mouse gross anatomy and development, timed","Bioinformatics operations, data types, formats, identifiers and topics","Flora Phenotype Ontology","Anatomical Entity Ontology","Kinetic Simulation Algorithm Ontology","Experimental Factor Ontology","MONDO: Monarch Disease Ontology","Agronomy Ontology","Molecular Process Ontology","Livestock Breed Ontology","Ancestry Ontology","Mouse pathology ontology","Mathematical Modelling Ontology","Ascomycete phenotype ontology","Neuro Behavior Ontology","nuclear magnetic resonance CV","Antibiotic Resistance Ontology","Mammalian Feeding Muscle Ontology","Animal Trait Ontology for Livestock","Minimum Information for A Phylogenetic Analysis (MIAPA) Ontology","Dictyostelium discoideum phenotype ontology","The Mammalian Phenotype Ontology","Mouse Developmental Stages","DICOM Controlled Terminology","Drug-drug Interaction and Drug-drug Interaction Evidence Ontology","MHC Restriction Ontology","Metabolomics Standards Initiative Ontology (MSIO)","Human Disease Ontology","NCBI organismal classification","Minimum PDDI Information Ontology","Ontology of Arthropod Circulatory Systems","The Data Use Ontology","OAE: Ontology of Adverse Events","Ontology of Biological Attributes","Evidence &amp; Conclusion Ontology (ECO)","Ontology for Biomedical Investigations","Ontology for General Medical Science","Ontology for BIoBanking (OBIB)","Plant Experimental Conditions Ontology","Ontology for genetic interval","planaria-ontology","Ontology of Genetic Susceptibility Factors (OGSF)","NCI Thesaurus OBO Edition","Ontology of Microbial Phenotypes","An ontology of core ecological entities","Plant Phenology Ontology","Ontology of Medically Related Social Entities","Scientific Evidence and Provenance Information Ontology","Obstetric and Neonatal Ontology","Non-Coding RNA Ontology","Human developmental anatomy, abstract","Social Insect Behavior Ontology","Mouse gross anatomy and development, timed","Ontology of Organizational Structures of Trauma centers and Trauma Systems","Semanticscience Integrated Ontology","eNanoMapper Ontology","Ontology of Biological and Clinical Statistics","Ensembl Glossary","OGG: Ontology of Genes and Genomes","Spider Ontology","Environment Ontology","Ontology for Parasite LifeCycle","STATO: the statistical methods ontology","Plant Environment Ontology","Plant Trait Ontology","Environment Ontology for Livestock","Orphanet Rare Disease Ontolog","Pathogen Transmission Ontology","Epidemiology Ontology","OVAE: Ontology of Vaccine Adverse Events","The Prescription of Drugs Ontology","eagle-i resource ontology","Unimod protein modification database for mass spectrometry","OHMI: Ontology of Host-Microbiome Interactions","Plant Ontology","EuPathDB ontology","Vertebrate trait ontology","Porifera (sponge) ontology","Medaka Developmental Stages","Exposure ontology","Vertebrate Taxonomy Ontology","Ontologized MIABIS","Regulation of Transcription Ontology","C. elegans development ontology","Ontology for MIRNA Target","Fungal gross anatomy","Regulation of Gene Expression Ontology","Biological Imaging Methods Ontology","C. elegans phenotype","The RNA Ontology (RNAO): An Ontology for Integrating RNA Sequence and Structure Data","Drosophila gross anatomy","Pathogen Host Interaction Phenotype Ontology","Rat Strain Ontology","FlyBase Controlled Vocabulary","Xenopus Anatomy Ontology","PRIDE Controlled Vocabulary","Name Reaction Ontology","Drosophila development","Pathway ontology","Experimental condition ontology","Sickle Cell Disease Ontology","Physico-chemical methods and properties","Cross-linker reagents ontology","Physico-chemical process","Influenza ontology","Sequence types and features ontology","Bioinformatics operations, data types, formats, identifiers and topics","Symptom Ontology","Food Ontology","Fission Yeast Phenotype Ontology","Units of measurement ontology","Tick Anatomy Ontology","microRNA Ontology","Combined phenotype ontology","Taxonomic rank vocabulary","Terminology for Description of Dynamics","Measurement method ontology","Variation Ontology","Mosquito gross anatomy ontology","Protein modification","Vaccine Ontology","Zebrafish anatomy and development ontology","C. elegans Gross Anatomy Ontology","HUPO-PSI cross-linking and derivatization reagents controlled vocabulary","Mass spectrometry ontology","The Oral Health and Disease Ontology","Phenotype And Trait Ontology","Population and Community Ontology","Platynereis Developmental Stages","PRotein Ontology (PRO)","Probability Distribution Ontology","OBO Relations Ontology","Systems Biology Ontology","Sustainable Development Goals Interface Ontology","Sample processing and separation techniques","Teleost taxonomy ontology","Uber-anatomy ontology","Zebrafish Experimental Conditions Ontology","Zebrafish developmental stages ontology"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Prefix<\/th>\n      <th>Namespace<\/th>\n      <th>Title<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}]}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="a-brief-rols-overview" class="section level1">
<h1 class="hasAnchor">
<a href="#a-brief-rols-overview" class="anchor"></a>A Brief rols overview</h1>
<p>The <em><a href="https://bioconductor.org/packages/3.9/rols">rols</a></em> package is build around a few classes that enable to query the OLS and retrieve, store and manipulate data. Each of these classes are described in more details in their respective manual pages. We start by loading the package.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="kw">library</span>(<span class="st">"rols"</span>)</a></code></pre></div>
<div id="ontologies" class="section level2">
<h2 class="hasAnchor">
<a href="#ontologies" class="anchor"></a>Ontologies</h2>
<p>The <code>Ontology</code> and <code>Ontologies</code> classes can store information about single of multiple ontologies. The latter can be easily subset using <code>[</code> and <code>[[</code>, as one would for lists.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">ol &lt;-<span class="st"> </span><span class="kw"><a href="../reference/Ontology-class.html">Ontologies</a></span>()</a>
<a class="sourceLine" id="cb3-2" data-line-number="2">ol</a></code></pre></div>
<pre><code>## Object of class 'Ontologies' with 217 entries
##    BCO, BFO ... ZECO, ZFS</code></pre>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="kw">head</span>(<span class="kw"><a href="../reference/Ontology-class.html">olsNamespace</a></span>(ol))</a></code></pre></div>
<pre><code>##    bco    bfo   bspo    bto   caro    cco 
##  "bco"  "bfo" "bspo"  "bto" "caro"  "cco"</code></pre>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1">ol[[<span class="st">"go"</span>]]</a></code></pre></div>
<pre><code>## Ontology: Gene Ontology (go)  
##   An ontology for describing the function of genes and gene products
##    Loaded: 2018-10-31 Updated: 2018-11-06 Version: 2018-10-30 
##    49722 terms  60 properties  0 individuals</code></pre>
<p>It is also possible to initialise a single ontology</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1">go &lt;-<span class="st"> </span><span class="kw"><a href="../reference/Ontology-class.html">Ontology</a></span>(<span class="st">"go"</span>)</a>
<a class="sourceLine" id="cb9-2" data-line-number="2">go</a></code></pre></div>
<pre><code>## Ontology: Gene Ontology (go)  
##   An ontology for describing the function of genes and gene products
##    Loaded: 2018-10-31 Updated: 2018-11-06 Version: 2018-10-30 
##    49722 terms  60 properties  0 individuals</code></pre>
</div>
<div id="terms" class="section level2">
<h2 class="hasAnchor">
<a href="#terms" class="anchor"></a>Terms</h2>
<p>Single ontology terms are stored in <code>Term</code> objects. When more terms need to be manipulated, they are stored as <code>Terms</code> objects. It is easy to obtain all terms of an ontology of interest, and the resulting <code>Terms</code> object can be subset using <code>[</code> and <code>[[</code>, as one would for lists.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1">gotrms &lt;-<span class="st"> </span><span class="kw"><a href="../reference/Term-class.html">terms</a></span>(go) ## or terms("go")</a>
<a class="sourceLine" id="cb11-2" data-line-number="2">gotrms</a></code></pre></div>
<pre><code>## Object of class 'Terms' with 49723 entries
##  From the GO ontology
##   GO:0005230, GO:0015276 ... GO:0032957, GO:0032942</code></pre>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1">gotrms[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>]</a></code></pre></div>
<pre><code>## Object of class 'Terms' with 10 entries
##  From the GO ontology
##   GO:0005230, GO:0015276 ... GO:0001819, GO:0044831</code></pre>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1">gotrms[[<span class="st">"GO:0090575"</span>]]</a></code></pre></div>
<pre><code>## A Term from the GO ontology: GO:0090575 
##  Label: RNA polymerase II transcription factor complex
##   A transcription factor complex that acts at a regulatory region of
##   a gene transcribed by RNA polymerase II.</code></pre>
<p>It is also possible to initialise a single term</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1">trm &lt;-<span class="st"> </span><span class="kw"><a href="../reference/Term-class.html">term</a></span>(go, <span class="st">"GO:0090575"</span>)</a>
<a class="sourceLine" id="cb17-2" data-line-number="2"><span class="kw"><a href="../reference/Term-class.html">termId</a></span>(trm)</a></code></pre></div>
<pre><code>## [1] "GO:0090575"</code></pre>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1"><span class="kw"><a href="../reference/Term-class.html">termLabel</a></span>(trm)</a></code></pre></div>
<pre><code>## [1] "RNA polymerase II transcription factor complex"</code></pre>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" data-line-number="1"><span class="kw">strwrap</span>(<span class="kw"><a href="../reference/Term-class.html">termDesc</a></span>(trm))</a></code></pre></div>
<pre><code>## [1] "A transcription factor complex that acts at a regulatory region of"
## [2] "a gene transcribed by RNA polymerase II."</code></pre>
<p>It is then possible to extract the <code>ancestors</code>, <code>descendants</code>, <code>parents</code> and <code>children</code> terms. Each of these functions return a <code>Terms</code> object</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb23-1" data-line-number="1"><span class="kw"><a href="../reference/Term-class.html">parents</a></span>(trm)</a></code></pre></div>
<pre><code>## Object of class 'Terms' with 1 entries
##  From the GO ontology
## GO:0044798</code></pre>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb25-1" data-line-number="1"><span class="kw"><a href="../reference/Term-class.html">children</a></span>(trm)</a></code></pre></div>
<pre><code>## Object of class 'Terms' with 41 entries
##  From the GO ontology
##   GO:0062071, GO:0008230 ... GO:0034718, GO:0030232</code></pre>
<p>Similarly, the <code>partOf</code> and <code>derivesFrom</code> functions return, for an input term, the terms it is a part of and derived from.</p>
<p>Finally, a single term or terms object can be coerced to a <code>data.frame</code> using <code>as(x, "data.frame")</code>.</p>
</div>
<div id="properties" class="section level2">
<h2 class="hasAnchor">
<a href="#properties" class="anchor"></a>Properties</h2>
<p>Properties (relationships) of single or multiple terms or complete ontologies can be queries with the <code>properties</code> method, as briefly illustrated below.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb27-1" data-line-number="1">trm &lt;-<span class="st"> </span><span class="kw"><a href="../reference/Term-class.html">term</a></span>(<span class="st">"uberon"</span>, <span class="st">"UBERON:0002107"</span>)</a>
<a class="sourceLine" id="cb27-2" data-line-number="2">trm</a></code></pre></div>
<pre><code>## A Term from the UBERON ontology: UBERON:0002107 
##  Label: liver
##   An exocrine gland which secretes bile and functions in metabolism
##   of protein and carbohydrate and fat, synthesizes substances
##   involved in the clotting of the blood, synthesizes vitamin A,
##   detoxifies poisonous substances, stores glycogen, and breaks down
##   worn-out erythrocytes[GO].</code></pre>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb29-1" data-line-number="1">p &lt;-<span class="st"> </span><span class="kw"><a href="../reference/Properties-class.html">properties</a></span>(trm)</a>
<a class="sourceLine" id="cb29-2" data-line-number="2">p</a></code></pre></div>
<pre><code>## Object of class 'Properties' with 112 entries
##  From the UBERON ontology
##   digestive system gland, abdomen element ... hepatobiliary system, bile</code></pre>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb31-1" data-line-number="1">p[[<span class="dv">1</span>]]</a></code></pre></div>
<pre><code>## A Property from the UBERON ontology: UBERON:0006925 
##  Label: digestive system gland</code></pre>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb33-1" data-line-number="1"><span class="kw"><a href="../reference/Term-class.html">termLabel</a></span>(p[[<span class="dv">1</span>]])</a></code></pre></div>
<pre><code>## [1] "digestive system gland"</code></pre>
</div>
</div>
<div id="use-case" class="section level1">
<h1 class="hasAnchor">
<a href="#use-case" class="anchor"></a>Use case</h1>
<p>A researcher might be interested in the trans-Golgi network. Searching the OLS is assured by the <code>OlsSearch</code> and <code>olsSearch</code> classes/functions. The first step is to defined the search query with <code>OlsSearch</code>, as shown below. This creates an search object of class <code>OlsSearch</code> that stores the query and its parameters. In records the number of requested results (default is 20) and the total number of possible results (there are 15190 results across all ontologies, in this case). At this stage, the results have not yet been downloaded, as shown by the 0 responses.</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb35-1" data-line-number="1"><span class="kw"><a href="../reference/OlsSearch-class.html">OlsSearch</a></span>(<span class="dt">q =</span> <span class="st">"trans-golgi network"</span>)</a></code></pre></div>
<pre><code>## Object of class 'OlsSearch':
##   query: trans-golgi network 
##   requested: 20 (out of 15190)
##   response(s): 0</code></pre>
<p>15190 results are probably too many to be relevant. Below we show how to perform an exact search by setting <code>exact = TRUE</code>, and limiting the search the the GO ontology by specifying <code>ontology = "GO"</code>, or doing both.</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb37-1" data-line-number="1"><span class="kw"><a href="../reference/OlsSearch-class.html">OlsSearch</a></span>(<span class="dt">q =</span> <span class="st">"trans-golgi network"</span>, <span class="dt">exact =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<pre><code>## Object of class 'OlsSearch':
##   query: trans-golgi network 
##   requested: 20 (out of 8)
##   response(s): 0</code></pre>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb39-1" data-line-number="1"><span class="kw"><a href="../reference/OlsSearch-class.html">OlsSearch</a></span>(<span class="dt">q =</span> <span class="st">"trans-golgi network"</span>, <span class="dt">ontology =</span> <span class="st">"GO"</span>)</a></code></pre></div>
<pre><code>## Object of class 'OlsSearch':
##   ontolgy: GO 
##   query: trans-golgi network 
##   requested: 20 (out of 755)
##   response(s): 0</code></pre>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb41-1" data-line-number="1"><span class="kw"><a href="../reference/OlsSearch-class.html">OlsSearch</a></span>(<span class="dt">q =</span> <span class="st">"trans-golgi network"</span>, <span class="dt">ontology =</span> <span class="st">"GO"</span>, <span class="dt">exact =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<pre><code>## Object of class 'OlsSearch':
##   ontolgy: GO 
##   query: trans-golgi network 
##   requested: 20 (out of 1)
##   response(s): 0</code></pre>
<p>One case set the <code>rows</code> argument to set the number of desired results.</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb43-1" data-line-number="1"><span class="kw"><a href="../reference/OlsSearch-class.html">OlsSearch</a></span>(<span class="dt">q =</span> <span class="st">"trans-golgi network"</span>, <span class="dt">ontology =</span> <span class="st">"GO"</span>, <span class="dt">rows =</span> <span class="dv">200</span>)</a></code></pre></div>
<pre><code>## Object of class 'OlsSearch':
##   ontolgy: GO 
##   query: trans-golgi network 
##   requested: 200 (out of 755)
##   response(s): 0</code></pre>
<p>Alternatively, one can call the <code>allRows</code> function to request all results.</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb45-1" data-line-number="1">(tgnq &lt;-<span class="st"> </span><span class="kw"><a href="../reference/OlsSearch-class.html">OlsSearch</a></span>(<span class="dt">q =</span> <span class="st">"trans-golgi network"</span>, <span class="dt">ontology =</span> <span class="st">"GO"</span>))</a></code></pre></div>
<pre><code>## Object of class 'OlsSearch':
##   ontolgy: GO 
##   query: trans-golgi network 
##   requested: 20 (out of 755)
##   response(s): 0</code></pre>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb47-1" data-line-number="1">(tgnq &lt;-<span class="st"> </span><span class="kw"><a href="../reference/OlsSearch-class.html">allRows</a></span>(tgnq))</a></code></pre></div>
<pre><code>## Object of class 'OlsSearch':
##   ontolgy: GO 
##   query: trans-golgi network 
##   requested: 755 (out of 755)
##   response(s): 0</code></pre>
<p>Let’s proceed with the exact search and retrieve the results. Even if we request the default 20 results, only the 8 relevant result will be retrieved. The <code>olsSearch</code> function updates the previously created object (called <code>qry</code> below) by adding the results to it.</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb49-1" data-line-number="1">qry &lt;-<span class="st"> </span><span class="kw"><a href="../reference/OlsSearch-class.html">OlsSearch</a></span>(<span class="dt">q =</span> <span class="st">"trans-golgi network"</span>, <span class="dt">exact =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb49-2" data-line-number="2">(qry &lt;-<span class="st"> </span><span class="kw"><a href="../reference/OlsSearch-class.html">olsSearch</a></span>(qry))</a></code></pre></div>
<pre><code>## Object of class 'OlsSearch':
##   query: trans-golgi network 
##   requested: 20 (out of 8)
##   response(s): 8</code></pre>
<p>We can now transform this search result object into a fully fledged <code>Terms</code> object or a <code>data.frame</code>.</p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb51-1" data-line-number="1">(qtrms &lt;-<span class="st"> </span><span class="kw">as</span>(qry, <span class="st">"Terms"</span>))</a></code></pre></div>
<pre><code>## Object of class 'Terms' with 8 entries
##  From  8 ontologies
##   NCIT:C33802, OMIT:0020822 ... GO:0005802, GO:0005802</code></pre>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb53-1" data-line-number="1"><span class="kw">str</span>(qdrf &lt;-<span class="st"> </span><span class="kw">as</span>(qry, <span class="st">"data.frame"</span>))</a></code></pre></div>
<pre><code>## 'data.frame':    8 obs. of  10 variables:
##  $ id                  : chr  "ncit:class:http://purl.obolibrary.org/obo/NCIT_C33802" "omit:class:http://purl.obolibrary.org/obo/OMIT_0020822" "go:class:http://purl.obolibrary.org/obo/GO_0005802" "cco:http://purl.obolibrary.org/obo/GO_0005802" ...
##  $ iri                 : chr  "http://purl.obolibrary.org/obo/NCIT_C33802" "http://purl.obolibrary.org/obo/OMIT_0020822" "http://purl.obolibrary.org/obo/GO_0005802" "http://purl.obolibrary.org/obo/GO_0005802" ...
##  $ short_form          : chr  "NCIT_C33802" "OMIT_0020822" "GO_0005802" "GO_0005802" ...
##  $ obo_id              : chr  "NCIT:C33802" "OMIT:0020822" "GO:0005802" "GO:0005802" ...
##  $ label               : chr  "Trans-Golgi Network" "trans-Golgi Network" "trans-Golgi network" "trans-Golgi network" ...
##  $ description         :List of 8
##   ..$ : chr "A network of membrane components where vesicles bud off the Golgi apparatus to bring proteins, membranes and ot"| __truncated__
##   ..$ : NULL
##   ..$ : chr "The network of interconnected tubular and cisternal structures located within the Golgi apparatus on the side d"| __truncated__
##   ..$ : chr "The network of interconnected tubular and cisternal structures located within the Golgi apparatus on the side d"| __truncated__
##   ..$ : chr "The network of interconnected tubular and cisternal structures located within the Golgi apparatus on the side d"| __truncated__
##   ..$ : chr "The network of interconnected tubular and cisternal structures located within the Golgi apparatus on the side d"| __truncated__
##   ..$ : chr "The network of interconnected tubular and cisternal structures located within the Golgi apparatus on the side d"| __truncated__
##   ..$ : chr "The network of interconnected tubular and cisternal structures located within the Golgi apparatus on the side d"| __truncated__
##  $ ontology_name       : chr  "ncit" "omit" "go" "cco" ...
##  $ ontology_prefix     : chr  "NCIT" "OMIT" "GO" "CCO" ...
##  $ type                : chr  "class" "class" "class" "class" ...
##  $ is_defining_ontology: logi  TRUE TRUE TRUE FALSE FALSE FALSE ...</code></pre>
<p>In this case, we can see that we actually retrieve the same term used across different ontologies. In such cases, it might be useful to keep only non-redundant term instances. Here, this would have been equivalent to searching the ncit, omit, go ontology</p>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb55-1" data-line-number="1">qtrms &lt;-<span class="st"> </span><span class="kw">unique</span>(qtrms)</a>
<a class="sourceLine" id="cb55-2" data-line-number="2"><span class="kw"><a href="../reference/Term-class.html">termOntology</a></span>(qtrms)</a></code></pre></div>
<pre><code>##  NCIT:C33802 OMIT:0020822   GO:0005802 
##       "ncit"       "omit"         "go"</code></pre>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb57-1" data-line-number="1"><span class="kw"><a href="../reference/Term-class.html">termNamespace</a></span>(qtrms)</a></code></pre></div>
<pre><code>## $`NCIT:C33802`
## NULL
## 
## $`OMIT:0020822`
## NULL
## 
## $`GO:0005802`
## [1] "cellular_component"</code></pre>
<p>Below, we execute the same query using the <em><a href="https://bioconductor.org/packages/3.9/GO.db">GO.db</a></em> package.</p>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb59-1" data-line-number="1"><span class="kw">library</span>(<span class="st">"GO.db"</span>)</a>
<a class="sourceLine" id="cb59-2" data-line-number="2">GOTERM[[<span class="st">"GO:0005802"</span>]]</a></code></pre></div>
<pre><code>## GOID: GO:0005802
## Term: trans-Golgi network
## Ontology: CC
## Definition: The network of interconnected tubular and cisternal
##     structures located within the Golgi apparatus on the side
##     distal to the endoplasmic reticulum, from which secretory
##     vesicles emerge. The trans-Golgi network is important in the
##     later stages of protein secretion where it is thought to play
##     a key role in the sorting and targeting of secreted proteins
##     to the correct destination.
## Synonym: Golgi trans face
## Synonym: Golgi trans-face
## Synonym: late Golgi
## Synonym: maturing face
## Synonym: TGN
## Synonym: trans face
## Synonym: trans Golgi network</code></pre>
</div>
<div id="on-line-vs-off-line-data" class="section level1">
<h1 class="hasAnchor">
<a href="#on-line-vs-off-line-data" class="anchor"></a>On-line vs. off-line data</h1>
<p>It is possible to observe different results with <em><a href="https://bioconductor.org/packages/3.9/rols">rols</a></em> and <em><a href="https://bioconductor.org/packages/3.9/GO.db">GO.db</a></em>, as a result of the different ways they access the data. <em><a href="https://bioconductor.org/packages/3.9/rols">rols</a></em> or <em><a href="https://bioconductor.org/packages/3.9/biomaRt">biomaRt</a></em> perform direct online queries, while <em><a href="https://bioconductor.org/packages/3.9/GO.db">GO.db</a></em> and other annotation packages use database snapshot that are updated every release.</p>
<p>Both approaches have advantages. While online queries allow to obtain the latest up-to-date information, such approaches rely on network availability and quality. If reproducibility is a major issue, the version of the database to be queried can easily be controlled with off-line approaches. In the case of <em><a href="https://bioconductor.org/packages/3.9/rols">rols</a></em>, although the load date of a specific ontology can be queried with <code>olsVersion</code>, it is not possible to query a specific version of an ontology.</p>
</div>
<div id="changes-in-version-2-0" class="section level1">
<h1 class="hasAnchor">
<a href="#changes-in-version-2-0" class="anchor"></a>Changes in version 2.0</h1>
<p><em><a href="https://bioconductor.org/packages/3.9/rols">rols</a></em> 2.0 has substantially changed. While the table below shows some correspondence between the old and new interface, this is not always the case. The new interface relies on the <code>Ontology</code>/<code>Ontologies</code>, <code>Term</code>/<code>Terms</code> and <code>OlsSearch</code> classes, that need to be instantiated and can then be queried, as described above.</p>
<table class="table">
<thead><tr class="header">
<th>version &lt; 1.99</th>
<th>version &gt;= 1.99</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><code>ontologyLoadDate</code></td>
<td>
<code>olsLoaded</code> and <code>olsUpdated</code>
</td>
</tr>
<tr class="even">
<td><code>ontologyNames</code></td>
<td><code>Ontologies</code></td>
</tr>
<tr class="odd">
<td><code>olsVersion</code></td>
<td><code>olsVersion</code></td>
</tr>
<tr class="even">
<td><code>allIds</code></td>
<td><code>terms</code></td>
</tr>
<tr class="odd">
<td><code>isIdObsolete</code></td>
<td><code>isObsolete</code></td>
</tr>
<tr class="even">
<td><code>rootId</code></td>
<td><code>olsRoot</code></td>
</tr>
<tr class="odd">
<td><code>olsQuery</code></td>
<td>
<code>OlsSearch</code> and <code>olsSearch</code>
</td>
</tr>
</tbody>
</table>
<p>Not all functionality is currently available. If there is anything that you need but not available in the new version, please contact the maintained by opening an <a href="https://github.com/lgatto/rols/issues">issue</a> on the package development site.</p>
</div>
<div id="cvparams" class="section level1">
<h1 class="hasAnchor">
<a href="#cvparams" class="anchor"></a>CVParams</h1>
<p>The <code>CVParam</code> class is used to handle controlled vocabulary. It can be used for user-defined parameters</p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb61-1" data-line-number="1"><span class="kw"><a href="../reference/CVParam-class.html">CVParam</a></span>(<span class="dt">name =</span> <span class="st">"A user param"</span>, <span class="dt">value =</span> <span class="st">"the value"</span>)</a></code></pre></div>
<pre><code>## [, , A user param, the value]</code></pre>
<p>or official controlled vocabulary (which triggers a query to the OLS service)</p>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb63-1" data-line-number="1"><span class="kw"><a href="../reference/CVParam-class.html">CVParam</a></span>(<span class="dt">label =</span> <span class="st">"MS"</span>, <span class="dt">accession =</span> <span class="st">"MS:1000073"</span>)</a></code></pre></div>
<pre><code>## [MS, MS:1000073, electrospray ionization, ]</code></pre>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb65-1" data-line-number="1"><span class="kw"><a href="../reference/CVParam-class.html">CVParam</a></span>(<span class="dt">label =</span> <span class="st">"MS"</span>, <span class="dt">name =</span><span class="st">"electrospray ionization"</span>)</a></code></pre></div>
<pre><code>## [MS, MS:1000073, electrospray ionization, ]</code></pre>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb67-1" data-line-number="1"><span class="kw"><a href="../reference/CVParam-class.html">CVParam</a></span>(<span class="dt">label =</span> <span class="st">"MS"</span>, <span class="dt">name =</span><span class="st">"ESI"</span>) ## using a synonym</a></code></pre></div>
<pre><code>## [MS, MS:1000073, ESI, ]</code></pre>
<p>See <code><a href="../reference/CVParam-class.html">?CVParam</a></code> for more details and examples.</p>
</div>
<div id="session-information" class="section level1">
<h1 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session information</h1>
<pre><code>## R Under development (unstable) (2018-11-09 r75569)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 14.04.5 LTS
## 
## Matrix products: default
## BLAS: /home/travis/R-bin/lib/R/lib/libRblas.so
## LAPACK: /home/travis/R-bin/lib/R/lib/libRlapack.so
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
## [1] DT_0.5               rols_2.11.0          GO.db_3.7.0         
## [4] AnnotationDbi_1.45.0 IRanges_2.17.1       S4Vectors_0.21.1    
## [7] Biobase_2.43.0       BiocGenerics_0.29.1  BiocStyle_2.11.0    
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_1.0.0         later_0.7.5        compiler_3.6.0    
##  [4] BiocManager_1.30.3 prettyunits_1.0.2  progress_1.2.0    
##  [7] tools_3.6.0        digest_0.6.18      bit_1.1-14        
## [10] jsonlite_1.5       evaluate_0.12      memoise_1.1.0     
## [13] RSQLite_2.1.1      pkgconfig_2.0.2    rlang_0.3.0.1     
## [16] shiny_1.2.0        DBI_1.0.0          crosstalk_1.0.0   
## [19] commonmark_1.6     curl_3.2           yaml_2.2.0        
## [22] pkgdown_1.1.0      xfun_0.4           httr_1.3.1        
## [25] stringr_1.3.1      roxygen2_6.1.1     xml2_1.2.0        
## [28] knitr_1.20         htmlwidgets_1.3    hms_0.4.2         
## [31] desc_1.2.0         fs_1.2.6           rprojroot_1.3-2   
## [34] bit64_0.9-7        R6_2.3.0           rmarkdown_1.10    
## [37] bookdown_0.7       blob_1.1.1         magrittr_1.5      
## [40] promises_1.0.1     backports_1.1.2    htmltools_0.3.6   
## [43] MASS_7.3-51.1      assertthat_0.2.0   xtable_1.8-3      
## [46] mime_0.6           httpuv_1.4.5       stringi_1.2.4     
## [49] crayon_1.3.4</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#introduction">Introduction</a><ul class="nav nav-pills nav-stacked">
<li><a href="#installation">Installation</a></li>
      <li><a href="#getting-help">Getting help</a></li>
      <li><a href="#the-resource">The resource</a></li>
      <li><a href="#the-package">The package</a></li>
      </ul>
</li>
      <li>
<a href="#a-brief-rols-overview">A Brief rols overview</a><ul class="nav nav-pills nav-stacked">
<li><a href="#ontologies">Ontologies</a></li>
      <li><a href="#terms">Terms</a></li>
      <li><a href="#properties">Properties</a></li>
      </ul>
</li>
      <li><a href="#use-case">Use case</a></li>
      <li><a href="#on-line-vs-off-line-data">On-line vs. off-line data</a></li>
      <li><a href="#changes-in-version-2-0">Changes in version 2.0</a></li>
      <li><a href="#cvparams">CVParams</a></li>
      <li><a href="#session-information">Session information</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Laurent Gatto.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
